[blog] 2021 survey results post (#30999)
mui/material-ui: docs/pages/blog/2021-developer-survey-results.md

---
title: 2021 MUI Developer Survey results
description: Results for our yearly developer survey, 2021 edition.
date: 2022-02-22T00:00:00.000Z
authors: ['danilo-leal']
tags: ['Developer survey']
---

> Keeping up with the tradition, a few months ago we opened the 2021 MUI Developer Survey, to which we received 1,589 responses, 7% more answers than last year. This time, we have almost doubled the survey size, adding more depth to each covered topic, especially opening space for more open-ended questions. We wholeheartedly appreciate everyone who contributed, you are all helping shape the future of MUI as a company, so, thank you!

Reviewer: We can sharpen up this intro by honing in on the value proposition for "you" the reader:

Why would I, as a developer who uses MUI, be interested in this information? (It's not always a easy question to answer!)

I would suggest framing it in these terms above all else: 

Your feedback helps us to build better products, and we can't thank you enough for being a part of our community. Nearly 1,600 of you responded to our 2021 Developer Survey, and we take your input very seriously. Here's what we've learned about your needs after poring over the results.

Just like the previous iterations, the survey was chopped into three sections: ["Your needs"](#your-needs), ["Your product"](#your-product), and ["About you"](#about-you).

## Your needs

### How would you feel if you could no longer use MUI?

> Fortunately, and similar to 2019 and 2020, over 93% of the respondents would be disappointed if they could no longer use MUI (a sum of _"Very disappointed"_ with _“Somewhat disappointed”_). A very slight 1% decrease from 2020 (94%) but still as encouraging as it can be. MUI’s team will continue to grow to keep you all even more satisfied.

Reviewer: Let's tighten this up to underscore the most important point: everybody loves us! 😁 I would also suggest replacing the third-person nouns with "you," and "MUI's team" with "we."

```suggestion
Over 93% of you would be disappointed if you could no longer use MUI (a sum of _"Very disappointed"_ and _“Somewhat disappointed”_). That's in keeping with 94% of you who felt the same way when we asked this question in 2019 and 2020. We'll continue to do our best to keep you satisfied over the long term!
```

<img src="/static/blog/2021-survey/1.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="Bar chart: 62.7% Very disappointed, 30.4% Somewhat disappointed, 6.9% Not disappointed." />

> <p class="blog-description">1567 out of 1589 answered</p>

Reviewer: Is it a style standard on the blog to omit periods from image descriptions like this?

Author: I guess not! I've omitted because Typeform's report does, so I thought it would make sense to follow it, but we're free to change it!

> On the flip side though, the amount of "Very disappointed” alone fell by 10,3%, and the number of "Not disappointed" grew by 1,1%, in comparison with 2020. When choosing the last option, respondents were able to write a free-form answer to justify the choice. Based on the data, here are the main reasons we noticed and possible explanations for why these came to be:

Reviewer: Suggestions for tightening up the text. I would also eliminate the decimals in the percentages as I don't think we need to be that precise here, or at least opt for the US English style of periods over commas: 10.3%.

```suggestion
Interestingly, the percentage who would be "Very disappointed" to stop using MUI fell by 10%, while those who would be "Not disappointed" grew by 1%. Those who chose the latter option were asked to explain why. Here are the biggest reasons we observed:
```

**The number of available alternatives**

> Lately, with every new sunrise, there's a new UI component library. Naturally, considering the amount of iteration each one does on the same problems, patterns start to become more and more established. It's common to see libraries with very similar implementations of components, styling solutions, theming, etc. It gets harder to find ways to differentiate, to find something that would make one's iteration more compelling. That's a main reason we do this survey yearly, to shed light into improvements opportunities. Now that we have uncovered a bunch of spaces to explore, and together with our steadily growing team, we hope to focus on more than one scope at once with the same level of attention to detail and quality we've strived to have.

Reviewer: Let's whittle this down to the main point: it's tough to stand out in this niche, but we're working on it.

```suggestion
Lately, with every new sunrise, there's a new UI component library. Because they aim to solve the same problems, these libraries often arrive at solutions that are very similar to one another. The establishment of common industrywide patterns makes it that much more difficult to stand out from the crowd. We're working hard to expand outward from our core products in order to provide you with tools that our competitors can't match.
```

**The introduction of a paid product**

> A big challenge with open-source is that the proportion of value captured by maintainers is close to zero. The MIT licensed model enabled a lot of growth but it naturally has reached a limit. In order to push our mission as much as we believe it's possible, we needed to introduce a way to capture more value. This was made by making MUI X open-core, instead of open-source, meaning having a full-time team of maintainers for the open-source and providing you with the most outstanding set of advanced components.

Reviewer 2: Maybe we should explain more what `open-core` means and emphasize that `DataGrid` is available under MIT license?
A lot of users tend to think that X components are commercial only (for example https://twitter.com/Bifrost60541029/status/1485648490902528004)

Author: ```suggestion
A big challenge with open-source is that the proportion of value captured by maintainers is close to zero. The MIT licensed model enabled a lot of growth but it naturally has reached a limit. In order to push our mission as much as we believe it's possible, we needed to introduce a way to capture more value. [Open-core](https://en.wikipedia.org/wiki/Open-core_model) allows us to still offer an MIT license (free) version of the X components while charging for a part of them, the one which demands more specialized attention (including support). 
```

Makes sense! Removed a bit of unnecessary text and swapped it for a brief mention of why open-core was relevant (linking a handy Wikipedia) and emphasizing there's still an MIT version of it.

Reviewer 2: Way to go!
I have rephrased it a bit, let me know what you think :)

> [Open-core](https://en.wikipedia.org/wiki/Open-core_model) allows us to still offer an MIT licensed (free) version of the X components while charging for additional features that require more attention (including support).

> It’s still the early days of MUI X, though. We introduced it at the end of 2020 and have been iterating on not only the components but on the model itself. We still have a lot more ground to cover, only ~0,1% of our developer community has been convinced to upgrade to the Pro plan so far. So, if you have more feedback about it, we'd be happy to hear. And if you want to understand more about our view of the MIT/open-source landscape, check our [Stewardship page](https://www.notion.so/Stewardship-542a2226043d4f4a96dfb429d16cf5bd).

Reviewer: ```suggestion
It’s still the early days of MUI X, though. We introduced it at the end of 2020 and have been iterating not only on the components but on the model itself. 

We still have a lot more ground to cover—only ~0.1% of our developer community has been convinced to upgrade to the Pro plan so far. Whether or not you've taken the plunge, we'd love to hear your thoughts about this. And if you want to understand more about our view of the MIT/open-source landscape, check our [Stewardship page](https://www.notion.so/Stewardship-542a2226043d4f4a96dfb429d16cf5bd).
```

> It’s still the early days of MUI X, though. We introduced it at the end of 2020 and have been iterating on not only the components but on the model itself. We still have a lot more ground to cover, only ~0,1% of our developer community has been convinced to upgrade to the Pro plan so far. So, if you have more feedback about it, we'd be happy to hear. And if you want to understand more about our view of the MIT/open-source landscape, check our [Stewardship page](https://www.notion.so/Stewardship-542a2226043d4f4a96dfb429d16cf5bd).

Author: Yup, agree!

**The breaking changes made in v5**

> Our most recent release, MUI Core v5, introduced some important breaking changes. The main problem we wanted to tackle with it was customizability, something we have been hearing as an improvement opportunity for a while. The changes done are meant to unlock more styling capacity while not deprioritizing performance. We understand that it introduced a considerable amount of work to migrate but, we’re still in the early days of maturing this new styling solution. There are a lot of opportunities to expand on the reason why the change was made, how to get the most out of it, best practices, tutorials, automation tools, and, generally, how-to guide those concerned about migrating to v5 because of it. If you have any ideas or suggestions, you’re always welcome to share your thoughts by opening an issue on our repository.

Reviewer: This is a very big block of text that could be split into several paragraphs for easier readability. When writing for online publications, I try to keep "paragraphs" to just 1, 2 or max 3 sentences. The goal is to make it as easy as possible to quickly skim.

```suggestion
Our most recent release, MUI Core v5, introduced some important breaking changes. The main problem we wanted to tackle was customizability: unlocking more styling options without compromising on performance. 

This new styling solution still has a ways to go in terms of maturity, and there's no denying that it requires a considerable amount of work to migrate. For our part, we still have a lot of work to do to help you make the most of these new features in v5, and we welcome your feedback. 

What can we do to help you succeed? If you have any ideas or suggestions, you’re always welcome to share your thoughts by opening an issue on our repository.
```

### How likely is it that you would recommend MUI to a friend or colleague?

> As for the Net Promoter Score, this year we also saw a decrease. Last year, we had 62,2% of the respondents as Promoters whereas now we had 56,4%. NPS itself moved from 62 in 2020 to 46 in 2021. If we consider the scale, we're still at the great level, though (30 to 70 is considered great), which makes us glad. We definitely have much more to do to achieve an excellent score. The reasoning for the previous and next question might fit to explain NPS data.

Reviewer: ```suggestion
This year we observed a decrease in Net Promoter Score (NPS): from 62 in 2020 to 46 in 2021. 56.4% of you were Promoters this year, down almost 6 percentage points from last year (62.2%). We're still in a good place—an NPS score between 30 and 70 is considered great. But we definitely have much more to do to achieve an excellent score.
```

<img src="/static/blog/2021-survey/2.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="Detractors: 10.88%, Neutrals: 32.69%, Promoters: 56.43%, Overall NPS: 46. Scale: -100 to 0 needs improvements, 0 to 30 is good, 30 to 70 is great, and 70 to 100 is excellent." />

<p class="blog-description">1563 out of 1589 answered</p>

### What is the main benefit you get from using MUI?

<img src="/static/blog/2021-survey/3.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="This question results in 2019 - Design: 248; Time: 193; DX: 128; Components: 89; Customizability: 53; Documentation: 25; Accessibility: 5; Typescript: 5; Community support: 4; This question results in 2021 - Time: 597; Design: 407; Components: 309; DX: 306; Customizability: 181; Documentation: 57; Community support: 26; Accessibility: 24; Performance: 17" />

<p class="blog-description">1422 out of 1589 answered this question in 2021</p>

> Here's what grew in terms of value proposition:

Reviewer: This could be more clear.

```suggestion
Here's what you liked the most about the product in 2021:
```

> - **Community (x2.9):** On a similar time frame, community size grew by x2.0. It looks like we're witnessing a network effect: with x2 more users, we got x3 more value from the community to the user base.

Reviewer: Is this a standard style choice to express the growth numbers as x2.9? Maybe this is a US vs. the world kind of thing, but it makes a lot more sense to me when these numbers are expressed as +290% for example. 

I would also expect to see "2x more developers" instead of "x2 more," but in this case I think I would opt for "twice" and "three times" instead.

```suggestion
- **Community (x2.9):** While appreciation for the community grew by 290%, the community itself doubled in size over that same time frame. It looks like we're witnessing a network effect: with twice as many users, we got three times more value from the community.
```

Author: Uhm, yeah. I think it's more an English vs Romance languages where the multiplier comes before and not after the number. That could be probably why for me (Portuguese) and Olivier (French) it didn't sound weird. But happy to go with the standard US English approach.

> - **Accessibility (x2.2):** Developers seem to be noticing the work we have put in, special thanks to Sebastian Silbermann [(@eps1lon)](https://github.com/eps1lon) and his continued help!

Reviewer 3: ❤️

> - **Accessibility (x2.2):** Developers seem to be noticing the work we have put in, special thanks to Sebastian Silbermann [(@eps1lon)](https://github.com/eps1lon) and his continued help!

Reviewer: ```suggestion
- **Accessibility (x2.2):** Developers seem to be noticing the work we have put in to make our components more accessible. Special thanks to Sebastian Silbermann [(@eps1lon)](https://github.com/eps1lon) for his continued help!
```

> - **Customizability (x1.6):** Apparently, you seem to be appreciating the new capabilities introduced in v5. However, it might still be far behind what we were aiming for, especially in light of the number of requests asking to better solve this problem.

Reviewer: ```suggestion
- **Customizability (x1.6):** Developers seem to be appreciating the new capabilities introduced in v5. That said, we recognize that there is still more work to be done here, given the number of requests we've gotten to better solve this problem.
```

> - **Components (x1.6):** The same goes for components, you seem to be enjoying the newly added components.

Reviewer: This one's tough, because the description really doesn't add any new information or insight. But what is there to say about this?

```suggestion
- **Components (x1.6):** We'll keep adding new components to the library as long as you continue to appreciate them this much!
```

Author: I guess that's good enough! It's meant only to state we're aware of the growing amount of requests for new components.

>   - Component quality (x1.0): It's encouraging to know you don't seem to be experience a decrease in quality. It's a risk we're taking with growing the MUI team, since it takes time for new team members to master the craft since we set the bar very high.

Reviewer: ```suggestion
  - Component quality (x1.0): You know that we set the bar very high for the quality of our components. That can be challenge as we continue to grow, because we expect nothing but the best from our team. Thankfully, this result shows that our new members have done an excellent job maintaining our standards.
```

> - **Time (x1.4):** This is an interesting one. It doesn't feel like MUI worked on this problem specifically. Could it be because the growing pressure it the market for going faster? Or because we somehow made this value proposition more noticeable? It could also be related to the next item.

Reviewer: ```suggestion
- **Time (x1.4):** This is an interesting one, because it's not an issue that we specifically addressed. Could this just be a result of the growing pressure in the market to build faster? Or because we somehow made this value proposition more noticeable? It could also be related to the next item.
```

- **Developer experience (x1.1):**

>   - Consistency (x1.75): We didn’t work on this dimension, and yet it grew. It's reasonable to assume it might be because we have more components, and then developers get to notice how consistent we make them be, highlighting this aspect.

Reviewer: ```suggestion
  - Consistency (x1.75): We didn’t work on this dimension, and yet it grew. It's reasonable to assume that because we have more components that ever before, developers have started to notice how much more consistent their work is when they use MUI.
```

>   - Ease of use (x1.0): It’s stable. Could it be because this is mostly defined by the React API (hooks) and/or because Material UI hasn’t made any major API changes that optimize for ease of use? There might be a lot of work on this dimension for our new developer experience team.

Reviewer: ```suggestion
  - Ease of use (x1.0): This remains unchanged. Could it be because ease of use is mostly defined by the React API (hooks), or is it because we haven't made any major API changes to optimize for ease of use? There could be a lot of work here for our new developer experience team.
```

And what has decreased:

> - **Design (x0.75):** We suffered from not onboarding a designer sooner.

Reviewer: "Onboarding" generally refers to the process of training and orienting a new hire, but what you're getting at here is that we didn't actually hire someone someone soon enough.

```suggestion
- **Design (x0.75):** We suffered from not bringing on a designer sooner.
```

  - Material Design (x0.4): Its selling power is clearly fading.

>   - Look & feel (x1.17): Surprising. This seems to be a transfer effect. It seems that people that used to answer Material Design now care more about the outcome that comes with it than the spec itself.

Reviewer: ```suggestion
  - Look & feel (x1.17): This is surprising. It seems to be a transfer effect:  people who previously cared more about Material Design now care more about the outcome than the spec itself.
```

    <br/>
    <br/>

<details>

>   <summary>See the broken down categories report.</summary>

Reviewer: "Broken down" is how you would describe a car that doesn't run anymore, contrasted with "breakdown," which describes separating information into smaller units.

```suggestion
  <summary>Click to see the breakdown of categories.</summary>
```

  <tr><td>3</td><td>components - data grid</td><tr>
  <tr><td>2</td><td>components - ?</td><tr>
  <tr><td>1</td><td>components - autocomplete</td><tr>
  <tr><td>1</td><td>components - base/unstyled</td><tr>
  <tr><th>306</th><th>DX</th><tr>
  <tr><td>221</td><td>DX - easy to use</td><tr>
  <tr><td>49</td><td>DX - consistency</td><tr>
  <tr><td>32</td><td>DX - API</td><tr>
  <tr><td>4</td><td>DX - ?</td><tr>
  <tr><th>181</th><th>customizability</th><tr>
  <tr><th>57</th><th>docs</th><tr>
  <tr><td>39</td><td>docs - ?</td><tr>
  <tr><td>17</td><td>docs - quality</td><tr>
  <tr><td>1</td><td>docs - quantity</td><tr>
  <tr><th>26</th><th>community</th><tr>
  <tr><th>24</th><th>accessibility</th><tr>
  <tr><th>17</th><th>performance</th><tr>
  <tr><td>15</td><td>performance - runtime</td><tr>
  <tr><td>1</td><td>performance - bundle size</td><tr>
  <tr><td>1</td><td>performance - ?</td><tr>
  <tr><th>9</th><th>community support</th><tr>
  <tr><th>5</th><th>icons</th><tr>
  <tr><th>3</th><th>typescript</th><tr>
  <tr><th>2</th><th>animations</th><tr>
</table>
</details>

### Please rate how much you agree or disagree with the following statements

> <img src="/static/blog/2021-survey/4.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="I can find most of the components I need (Strongly agree: 45.5%, Agree: 45.9%, Neutral: 6.2%, Disagree: 1.5%, Strongly Disagree 0.8%); I can easily customize the components to match the desired design (Strongly agree: 23.4%, Agree: 46.1%, Neutral: 18.6%, Disagree: 9%, Strongly Disagree 2.9%); I can find the answers to most of my questions in the documentation (Strongly agree: 24.1%, Agree: 50.1%, Neutral: 16.3%, Disagree: 7.6%, Strongly Disagree 1.9%); I find the library's performance to be great (Strongly agree: 26.4%, Agree: 44.3%, Neutral: 22.6%, Disagree: 4.7%, Strongly Disagree 2%); Whenever I needed to get help, I received helpful responses, StackOverflow or GitHub (Strongly agree: 20.9%, Agree: 36.7%, Neutral: 36%, Disagree: 4.6%, Strongly Disagree 1.8%)" />

Author: Oh yes, noted! Good call.

<p class="blog-description">1534 out of 1589 answered</p>

> It's definitely a good thing to have a general sense of agreement to most of the statements. However, there's a considerably big gap between _Strongly agree_ and _Agree_ for all but the first statement, which showcases that we need to continue pushing for even more quality in each of these dimensions (customizability, customization, performance, and support).

Reviewer: I believe this is the first time in the piece where the potential responses are italicized. Do we want to remove the italics here, or should we instead italicize them throughout?

Author: I have italicized a few other instances just for consistency's sake. Is there a more correct approach?

### What are your most important criteria for choosing a UI library?

<img src="/static/blog/2021-survey/5.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="Ranked list: 1. The design - look and feel; 2. Customizability; 3. Documentation quality; 4. Comprehensiveness; 5. Performance; 6. Popularity; 7. Accessibility; 8. Offered support and help; 9. Bundle size" />

> <p class="blog-description">1500 out of 1589 answered - Check <a href="https://www.typeform.com/help/a/ranking-question-360052767651/">Typeform's documentation</a> to understand ranking average calculations.</p>

Reviewer: Remove trailing period in keeping with the style of the piece.

```suggestion
<p class="blog-description">1500 out of 1589 answered - Check <a href="https://www.typeform.com/help/a/ranking-question-360052767651/">Typeform's documentation</a> to understand ranking average calculations</p>
```

> This year we experimented using Typeform to run the survey, which allowed developers to rank the criteria available - something that wasn't possible with Google Forms. That way, we didn't have to limit the answers to 3 options, allowing to capture a richer perspective. But, even so, the results weren’t too different from last year. Design (look and feel) remains the most important criterion for choosing UI libraries, and customization and documentation quality are still there in the top 3. The notable difference this year seems to be performance climbing its way up, as it has been ranked as the fifth most important aspect.

Reviewer: ```suggestion
This year we experimented with using Typeform to run the survey, which allows respondents to rank their choices. This could've given us a richer perspective than we've been able to capture in the past, but in the end the results weren't too different from last year. 

Design (look and feel) remains the most important criterion for choosing UI libraries, and customization and documentation quality are still there in the top 3. The most notable difference seems to be performance, which climbed to number 5 this year.
```

### What else can we do to improve MUI for you?

<img src="/static/blog/2021-survey/6.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="Word cloud of what we can do to improve MUI" />

<p class="blog-description">1007 out of 1589 answered this question in 2021</p>

> Here are the most recurring topics:

Reviewer: ```suggestion
Here are the most recurring topics for improvement:
```

> - **More components:** We continue to see requests for more advanced components, following last year's pattern, such as Charts, Forms, and Calendar, to name a few.

Reviewer: ```suggestion
- **More components:** We continue to see requests for more advanced components such as charts, forms, and calendars, to name a few.
```

> - **More examples:** Knowing the "officially recommended" ways of using any product is always helpful. We know that many of the learning resources you find out there right now are outdated (based on v4), so that's absolutely worth of investing.

Reviewer: ```suggestion
- **More examples:** Many of the existing learning resources out there are now outdated following the v5 upgrade, so we have a lot of work to do to show you how to make the most of our products.
```

> - **Provide more themes:** Even with the launch of Material v3, Material Design has long been a design language that some find outdated. That's why we are working on a second design system, thought to offer everything that the current Material package does (component quantity and quality) but with an alternative design direction.

Reviewer: ```suggestion
- **Provide more themes:** Even with the launch of Material v3, Material Design is seen as outdated by many. That's why we're working on a second design system that will offer the same high standard of quality as Material, but with an alternative design direction.
```

> - **Fewer breaking changes:** MUI Core v5 definitely introduced some important breaking changes, especially because of the new styling solution. We’ll continue to explore the v5 migration topic across the post but, for now, it's good to share that we don’t expect to release any major this year. In fact, we aim for majors happening with at least 12 months apart from each other.

Reviewer: ```suggestion
- **Fewer breaking changes:** MUI Core v5 definitely introduced some important breaking changes, especially because of the new styling solution. Rest assured that we don’t expect to release any major updates this year—in fact, we aim to keep majors at least 12 months apart from each other.
```

> - **Improve customization:** Many of the entries for customization ask for making it easier, to add more examples of common customization use-cases (like font-family and primary/secondary color change), and improvements to theme capabilities. Even with the popularity of emotion and styled-components, there continues to be an enourmous need to facilitate customization.

Reviewer: Some grammar fixes here. Also, Emotion gets capitalized, I believe.

```suggestion
- **Improve customization:** Common requests include making customization easier, providing more examples of common use-cases (like font-family and primary/secondary colors), and improving to theme capabilities. Even with the popularity of Emotion and styled-components, there continues to be an enormous need to facilitate customization.
```

> If you have requests or ideas for improving the above-mentioned topics, here are a few tips to have them prioritized faster:

Reviewer: "Above-mentioned" is rather clunky; "aforementioned" would be more fitting, but I think we can be more clear here.

```suggestion
Got ideas for improvements? Please share them with us! Here's how to make sure that your requests get top priority:
```

> - In both MUI Core and X repositories, you should label a request issue with `Waiting for upvotes`. If your issue gains a significant amount of upvotes, it will definitely get our attention and consideration.

Reviewer 2: > you should label a request issue

Maybe we can rephrase this part, Since users cannot label issues?

> - In both MUI Core and X repositories, you should label a request issue with `Waiting for upvotes`. If your issue gains a significant amount of upvotes, it will definitely get our attention and consideration.

Author: ```suggestion
- In both MUI Core and X repositories, we'll label issues with feature/component requests with `Waiting for upvotes`. Make sure to structure, research and justify the request so it resonates with more developers, and, eventually, gets enough priority to be implemented.
```

I tend to forget the community doesn't have the same permissions as we do 😅 Here's a rephrased version considering both of your points. Let me know what you think!

> - In both MUI Core and X repositories, you should label a request issue with `Waiting for upvotes`. If your issue gains a significant amount of upvotes, it will definitely get our attention and consideration.

Reviewer 2: Looks great!

> - In both MUI Core and X repositories, you should label a request issue with `Waiting for upvotes`. If your issue gains a significant amount of upvotes, it will definitely get our attention and consideration.

Reviewer: Suggesting some revisions for grammar and clarity:

-In both MUI Core and X repositories, we'll label issues requesting features or components as `Waiting for upvotes`. So, make sure to structure, research and justify the request so it resonates with more developers, and, eventually, gets enough priority to be implemented.

+When you create an issue to request features or components in the MUI Core or X repositories, we'll label it as `Waiting for upvotes`. The more votes your issue receives, the higher a priority it will become—so make sure to structure, research, and justify your request so it resonates with other community members.

> - When requesting more components, make sure to benchmark existing implementations of it in the wild as it’s very helpful to see where they succeed and fail, as upfront learning content. It’s also great to explain as best as you can what is the problem you’re trying to solve. Oftentimes, an already available component might already solve it.

Reviewer: singular/plural misalignment. Also, I think this is really two different points: show us some benchmarks on the one hand, and clearly describe your problem on the other hand. In fact, I might rearrange this list so that the latter point is the first one, above the point about upvotes, because it negates all other points if it turns out the component they need already exists.

```suggestion
- When requesting a new component, make sure to benchmark existing implementations of it in the wild so we can see what works and what we could better. 

- Try to explain the problem you're having as clearly as possible—it's often the case that we already have a component to solve it.
```

> - If you're requesting easier customization, make sure to demonstrate the desired outcome and explain as detailed as you can the doubts you're facing. Very frequently these questions showcase opportunities to increase documentation quality.

Reviewer: I'm not sure that "doubts" is the right word here.

```suggestion
- If you're requesting easier customization, make sure to demonstrate the desired outcome, and explain where you're having trouble in as much detail as you can. These requests often highlight opportunities for us to improve our documentation.
```

## Your product

### What were you primarily using before MUI?

<img src="/static/blog/2021-survey/7.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="Bootstrap: 40.6%; Started with MUI: 37.4%; Tailwind:: 4.8%; Ant Design: 4.8%; Angular Material: 4%; Semantic-UI: 4%; Other: 3.8%; Chakra UI: 0.7%" />

<p class="blog-description">1389 of 1589 answered</p>

> Not too different from last year with the exception that, this time, we’ve seen way fewer developers claiming to be using custom systems before. Very encouraging to know the number of people starting their products off with MUI!

Reviewer: Incomplete sentences.

```suggestion
These results are not too different from last year, with the exception that far fewer developers reported using custom system before MUI. It's very encouraging to know that so many of you are building your products with MUI right from the start!
```

### Are you using any of these in addition to MUI?

<img src="/static/blog/2021-survey/8.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="Nope, just MUI: 70.7%; Tailwind: 10.3%; Bootstrap: 9.8%; Ant Design: 4%; Semantic-UI: 1.7%; Other: 1.6%; Chakra UI: 1.2%; Angular Material: 0.8%" />

<p class="blog-description">1468 out of 1589 answered</p>

> We're glad to see that for many developers MUI seems to be covering all the needs as far as component UI libraries are concerned. This is definitely something we optimize for.

Reviewer: We can eliminate the "weasel words" here for a stronger statement overall.

```suggestion
We're glad to see that MUI largely covers all of your needs for a component UI library—this is a major priority for us.
```

### Who are you building for?

<img src="/static/blog/2021-survey/9.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="For the company I work at: 65.6%; For my personal side project: 20.8%; For a client: 12.9%; Other: 0.7%" />

<p class="blog-description">1523 out of 1589 answered</p>

MUI for client work and for personal use have swapped places this year. Maybe v5 got developers interested enough to try it on their own sites and apps? Interesting to see.

### Who do you collaborate most with?

<img src="/static/blog/2021-survey/10.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="Other developers: 67.2%; Designers: 34.3%; Product managers: 29.2%; No one, it's just me: 28.9%; Other: 0.9%" />

<p class="blog-description">1527 out of 1589 answered</p>

> We didn’t ask this question on the previous iteration but thought that doing so now would be a good idea, given we know that our documentation is not only visited by developers. Optimizing our language, and even the code itself, to be understandable by other team members such as designers and product managers makes the whole design system conversation way more inclusive and approachable. And the results illustrate that this thinking might make a lot of sense indeed.

Reviewer: We can sharpen this quite a bit by getting straight to the point.

```suggestion
This is a new question in our annual survey, and the results highlight the fact that our documentation is not just for developers. It's crucial for our docs—and even the code itself—to be easily understood by designers, product managers, and other less technical team members.
```

### How many web applications did you or your team deliver using MUI this year?

<img src="/static/blog/2021-survey/11.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="0-1: 494; 2-3: 381; 4-5: 122; 6-10: 26; 10+: 28" />

<p class="blog-description">1051 out of 1589 answered</p>

> When discussing potential indicators we could track, perhaps one of the hardest ones is efficacy - or, in other words, how much does MUI actually help shipping products? The only way to properly know, since this is usually private, is by asking developers. The number of apps delivered is, of course, contextual, but, regardless, it's encouraging to know that we might be enabling these applications to be built faster than they would've if developers were to build it all from scratch.

Reviewer: ```suggestion
Efficacy is one of the toughest metrics to track—how much does MUI actually help developers in shipping their products? The number of apps delivered may vary widely between teams and industries, but it's still encouraging to see that we are enabling developers to build much faster than if they started from scratch.
```

### Which MUI products do you use in your application?

<img src="/static/blog/2021-survey/12.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="MUI Core: MIT licensed foundational components, right now available with Material Design: 96.6%; MUI X: collection of advanced components available under MIT and commercial licenses: 14.7%; MIT License: 126 responses; Commercial licenses: 99 responses" />

<p class="blog-description">1551 out of 1589 answered</p>

> Definitely not a very surprising answer, we know that most of our user base is here for MUI Core. The proportion of MUI X users in the MIT and commercial license might not be true, though. We emailed those who bought the MUI X commercial license asking to answer the survey, so they had this plus one ad of it.

Reviewer: When you say "true" here, are you suggesting that the results may not be accurate since these respondents got pinged twice? 

Even so, I don't think it's really worth bringing up. Most people need to be emailed 7 or 8 times about a product or service before they take action. We should celebrate the win instead! And it's an opportunity to point out where the business is headed, and why that's cool for our users. :)

```suggestion
It's not surprising to find out that most of our user base is here for MUI Core. But it is very encouraging for us to see such a high proportion of MUI X users operating with commercial licenses, and we are excited to continue expanding on our paid products for these users.
```

> However, there is a lot of opportunities in the MUI X space to unpack, let's explore a little bit further below.

Reviewer: ```suggestion
There are a lot of opportunities in the MUI X space to unpack. Let's explore a little bit further below.
```

### Did you know what MUI X was prior to this survey?

<img src="/static/blog/2021-survey/13.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="Yes: 54.5%; No: 45.5%" />

<p class="blog-description">1312 out of 1589 answered</p>

A large chunk of you didn’t know what MUI X is prior to the survey, which makes it clear that we still have a lot to do to expand its reach.

### Are you currently using any paid UI component library?

<img src="/static/blog/2021-survey/14.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="Yes: 11.8%; No: 88.2%" />

<p class="blog-description">1584 out of 1589 answered</p>

> The results are not very different from last year, aside from a slight increase of respondents saying yes. The majority of respondents don’t use any but out of those who do, it is mostly MUI X, which is great news.

Reviewer: ```suggestion
The results are not very different from last year, aside from a slight increase of respondents saying yes. The majority of respondents don’t use any paid libraries, but those who do are mostly using MUI X, which is great news.
```

> ### How can we improve the Data Grid for you?

Reviewer: What do we think about the capitalization of Data Grid? Our docs primarily refer to individual components in lowercase.

Author: How about we continue discussing it in #31101 and then bring the conclusion back here?!

Reviewer: Good call!

<img src="/static/blog/2021-survey/15.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="Customizability: 21.9%; Cheaper Pro plan: 17.2%; More features: 12.5%; Fix features: 9.4%; Improve docs: 6.3%; Look & feel: 6.3%; Maintain it: 4.7%; Bugs: 3.1%; DX: MUI Core consistency: 3.1%; Performance: 3.1%; Transition: 3.1%; SSR: 1.6%" />

<p class="blog-description">64 out of 1589 answered</p>

> - **Customizability:** We already touched on this topic before and the frequency that it shows up on the MUI X space is a definitely a hint to double down on efforts to make customizations easier to do. We're aware that we might be missing some stuff, like documention around theming and a headless API.

Reviewer: ```suggestion
- **Customizability:** This is a common topic for us, and it comes up often when discussing MUI X. Your feedback tells us that we need to double down on our efforts to make customization easier. We also recognize that we are still lacking in some important documentation around theming and headless APIs.
```

> - **Cheaper Pro plan:** We know it can be out of range for a lot of developers and that's why the free version is already packed with many features. MUI mainly develops OSS software and we needed a way to sustain the project longer, hence the price we’re currently exercising, which actually is, still, way lower than similar products. However, many who left feedback around the price are individuals, so there might be an opportunity to evolve it for those not backed by their company.

Reviewer: I don't think there's any need to try to apologize for our price point, or explain why we need money to operate. Instead, this is an opportunity to point out how awesome OSS is!

```suggestion
- **Cheaper Pro plan:** We know it can be out of range for a lot of developers, and that's why the free version is already packed with many features. MUI mainly develops open-source software, and we deeply value the OSS ethos of freely sharing what we build so that others can use it and improve upon it.

The price of our Pro plan positions it as a product intended for use by teams at professional organizations. That said, many who left feedback on the price are individuals, so there might be an opportunity to evolve it for those not backed by their company.
```

> - **More features & fix features:** Collapsable rows, column resizing, and features for ERP apps were some of the features requested. And regarding features to fix, pagination with REST API, backend filtering, and cell editor, were the main mentioned.

Reviewer: ```suggestion
- **More features & feature fixes:** Collapsable rows, column resizing, and features for ERP apps were some of the features requested. And regarding features to fix, the most common requests were pagination with REST APIs, backend filtering, and cell editor.
```

> - **Improvements to the documentation:** Basically similar to the already discussed aspects of documentation improvement. Data Grid’s docs today could certainly use a better and optimized space and that’s we’ve been working on, so it’s easier to find and expand on the amount of information.

Reviewer: ```suggestion
- **Improvements to the documentation:** The data grid docs could certainly use a major overhaul—we're working on it.
```

> - **Improvements to the look and feel:** It might be probably a solid idea to think of different default designs to offer on the Data Grid, alternative to Material Design. On a more UX side of things, we've heard many developers mentioning the filtering experience specifically.

Reviewer: This bullet point contains a lot of weasel words that can be removed to make a stronger point.

```suggestion
- **Improvements to the look and feel:** Data grid builders want more design options beyond Material. In terms of UX, many developers mention the filtering experience specifically.
```

### How can we improve the Data Grid Pro for you?

> And now, here's what you mentioned about making the commercially licensed version of the Data Grid better:

Reviewer: This can be cut.

<img src="/static/blog/2021-survey/16.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="More features: 39.8%; Customizability: 20.4%; Fix features: 15.1%; Performance: 5.4%; Cheaper Premium plan: 3.2%; Improve docs: 3.2%; DX - API: 2.2%; LGPL license: 2.2%; More components: 2.2%" />

<p class="blog-description">75 out of 1589 answered</p>

> - **More features:** [Grouping](https://mui.com/components/data-grid/group-pivot/#row-grouping) was the most requested feature for the Data Grid Pro, followed by [master detail](https://mui.com/components/data-grid/group-pivot/#master-detail), aggregation, search, [tree data](https://mui.com/components/data-grid/group-pivot/#tree-data), and [column pinning](https://mui.com/components/data-grid/columns/#column-pinning). Good thing is that we released some of these already during the time we were anlayzing the survey results. Check the links to see their docs and demos. Alternatively, you can follow [our public roadmap](https://github.com/mui/mui-x/projects/1) to see when the others will land.

Reviewer: If "data grid" is lowercase but "Pro" plan gets a capital, then how do we feel about "Data Grid Pro"?

```suggestion
- **More features:** [Grouping](https://mui.com/components/data-grid/group-pivot/#row-grouping) was the most requested feature for the Data Grid Pro, followed by [master detail](https://mui.com/components/data-grid/group-pivot/#master-detail), aggregation, search, [tree data](https://mui.com/components/data-grid/group-pivot/#tree-data), and [column pinning](https://mui.com/components/data-grid/columns/#column-pinning). We've already released some of these in the time it took us to analyze the survey results. Check the links to see their docs and demos. Alternatively, you can follow [our public roadmap](https://github.com/mui/mui-x/projects/1) to see when the others will land.
```

> - **Customizability:** The majority of requests about customizability were actually asking for possibilities to tweak Data Grid interactions, which is something we need to understand more exactly what it means. Following next were requests for more styles customizability and documentation around how to do that (similar to the discussed above).

Reviewer: ```suggestion
- **Customizability:** The majority of requests about customizability were asking for possibilities to tweak data grid interactions, which we need to delve into more deeply to better understand. Following next were requests for more style customizability, and more documentation on customization.
```

- **Fix features:** Filtering got first place as the most requested feature to fix on the Data Grid Pro. Following next were lazy loading and server-side rendering support.

> - **Performance:** Requests about performance were mostly spread out between runtime and bundle size. We acknowledge we have done almost no effort on the bundle size so far. There are likely some low-hanging fruits to reduce it.

Reviewer: ```suggestion
- **Performance:** Requests about performance were mostly spread out between runtime and bundle size. We must admit that we have put almost no effort into optimizing the bundle size so far. There are likely some low-hanging fruits there.
```

> ### What were you using before the Data Grid?

Reviewer: ```suggestion
### What were you using before the data grid?
```

<img src="/static/blog/2021-survey/17.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="In house: 29.9%; MUI Table: 17.9%; Nothing: 7.5%; material-table: 6%; material-datatables: 6%; Sencha: 6%; Kendo UI: 4.5%; react-table: 4.5%; Bootstrap Table: 3%; AG Grid: 3%; datatables.net: 3%; JSP: 1.5%" />

<p class="blog-description">149 out of 1589 answered</p>

> To finish off the section about MUI X, it's interesting to see that a lot of you are taking the effort yourselves to build Data Grid alike tables. We'd love to understand more why it made sense to build it in-house, especially given we know it's no trivial work at all. Also, it's refreshing to see that the standard MUI table components is already of good help for a lot of you.

Reviewer: ```suggestion
It's interesting to see that a lot of you are building your own in-house data grids. We'd love to know more about why your team made this choice, especially because we know exactly how much work it can be. Also, it's refreshing to see that the standard MUI table components are adequate for many of you.
```

### What are you building?

<img src="/static/blog/2021-survey/18.png" style="width: 796px; margin-top: 16px; margin-bottom: 16px;" alt="Dashboard admin applications: 26.3%; Enterprise applications: 25.9%; A custom design system: 9.7%; Landing page: 8.1%; E-commerce application: 7.8%; Personal website/portfolio: 6.9%; CMS: 6.1%; Just playing with tech: 4.6%; Blog: 2.8%; Other: 1.4%" />

> Enterprise, dashboards applications, and design systems still continue to be in the top 3, similar to last year. The slight changes though are developers using MUI more to build e-commerce apps and portfolios, which is pretty cool. We'd love to see them, feel free to share in the ["Who's using Material UI?" issue](https://github.com/mui/material-ui/issues/22426).

Reviewer: ```suggestion
Enterprise, dashboard applications, and design systems continue to be in the top 3, similar to last year. What's changed is that developers are now using MUI more to build e-commerce apps and portfolios, which is pretty cool. We'd love to see them! Feel free to share in the ["Who's using Material UI?" issue](https://github.com/mui/material-ui/issues/22426).
```

### What delivery mechanisms are you using?

<img src="/static/blog/2021-survey/19.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="Single-page app (Create React App, etc): 74.7%; Server-side rendered website (Next.js, Gastby, etc): 20.8%; Desktop app (Electron, etc): 3.5%; Native mobile app (Cordova, etc): 0.6%; Other: 0.4%" />

<p class="blog-description">1509 out of 1589 answered</p>

### What type system are you using?

<img src="/static/blog/2021-survey/20.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="TypeScript: 63.8%; None: 18%; prop-types: 16.6%; Flow: 1.4%; Other: 0.2%" />

<p class="blog-description">1501 out of 1589 answered</p>

> Very different from last year, Typescript just jumped off and it's the primary type system used out there. It seems reasonable to expect more people to use a type system rather than not.

Reviewer: Incomplete sentence.

```suggestion
These results are very different from last year—Typescript just jumped off and it's now the primary type system used out there. It seems reasonable to expect more people to start using type systems.
```

### Which framework are you using, if any?

<img src="/static/blog/2021-survey/21.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="Create React App: 62.4%; Next.js: 21.9%; Custom webpack: 10.7%; None: 2.6%; Gatsby: 0.9%; Other: 1.5%" />

<p class="blog-description">1497 out of 1589 answered</p>

> A considerable bump in Next.js usage, which is not surprising due to the popularity and advancements it gained in 2021.

Reviewer: Incomplete sentence.

```suggestion
This year we saw a considerable bump in Next.js usage, which is not surprising due to its increasing popularity and the advancements it gained in 2021.
```

### What styling solution are you using?

<img src="/static/blog/2021-survey/22.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="MUI Core v4 (JSS): 45%; Styled components: 37.9%; Emotion: 30.2%; SASS: 20.8%; CSS modules: 18.9%; Vanilla CSS: 17.6%; Tailwind CSS: 9.1%; Stitches: 0.4%; Other: 1.2%" />

<p class="blog-description">1492 out of 1589 answered</p>

> As the MUI Core v5 release is still relatively recent, it's still expected to see a great number of developers using JSS, and there will always probably be some of you who use it because of pure preference.

Reviewer: ```suggestion
As the MUI Core v5 release is relatively recent, we expect to see many developers still using JSS, and there will probably always be some of you who will prefer it.
```

> However, it’s notable the styled-components and emotion growth, and that might be because it’s what our new styling solution is based off. We're still seeing a lot of questions and also opportunities to make it better, so this year we'll focus on expanding and refining [MUI System](https://mui.com/system/basics/) even further.

Reviewer: ```suggestion
But the growth of Emotion and styled-components here is noteworthy, because they are what our new styling solution is based on. This year we'll be focusing on expanding and refining [MUI System](https://mui.com/system/basics/) to further improve upon this new styling solution.
```

### Did you recently migrate to MUI Core v5?

<img src="/static/blog/2021-survey/23.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="Yes: 63.3%; No: 36.7%" />

<p class="blog-description">1546 out of 1589 answered</p>

> Even though a lot of you who answered the survey has migrated, that are still a lot more to convice migrating. 2022 is going to be a lot focused in that.

Reviewer 2: ```suggestion
Even though a lot of you who answered the survey has migrated, that are still a lot more to convince migrating. 2022 is going to be a lot focused in that.
```

> Even though a lot of you who answered the survey has migrated, that are still a lot more to convince migrating. 2022 is going to be a lot focused in that.

Reviewer: ```suggestion
Though many of you who answered the survey have already migrated, we still have a many more to convince to come along with us. That'll be a major focus of ours for the year ahead.
```

### What's the statement that most defines your migration experience?

<img src="/static/blog/2021-survey/24.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="It was tricky but MUI's documentation and resources helped me to get it right: 43%; It was smooth and it is now working fine: 38.5%; It wasn't good, I had many problems and it took me a long time to finish: 14.3%; It was bad and I ended up regretting it: 2.8%; Other: 1.4%" />

<p class="blog-description">930 out of 1589 answered</p>

### What could MUI do to improve the migration experience?

<img src="/static/blog/2021-survey/25.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="General documentation improvements: 43.2%; Automation: 20.7%: Fewer breaking changes: 15.6%; Fewer styling soluiton breaking changes: 8.5%; More tutorials: 5.4%; More styling solution migration tutorial: 4.4%; Migrate Core and X together: 1%" />

<p class="blog-description">472 out of 1589 answered</p>

> - **General documentation improvement:** Documentation plays a huge part in defining the developer experience. We definitely need to explore more ways of making the step-by-step easier, add more examples, refine the copywriting, and maybe overall make it more beginner-friendly.

Reviewer: ```suggestion
- **General documentation improvement:** Documentation plays a huge part in defining the developer experience. We definitely need to refine our copywriting, provide more examples, and offer more beginner-friendly resources.
```

>   - There are other topics converging with this one, like requests for more tutorials.

Reviewer: This line can be cut for redundancy.

> - **Automation:** We prepared codemods to aid the v5 migration, given the amount of changes it had. We've seen positive feedback for it, which is encouraging since it's our first time providing tools to automate a migration process. However, it is becoming clear how and when the codemods weren't too effective, so we'll make sure to refine it more.

Reviewer: ```suggestion
- **Automation:** We prepared codemods to help with the v5 migration, given the number of changes. This has been met with positive feedback, which is encouraging since it's our first time providing tools to automate the migration process. However, it is becoming clear how and when the codemods weren't too effective, so we'll make sure to refine it more.
```

### Could you share the reason why you haven't migrated yet?

<img src="/static/blog/2021-survey/26.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="No bandwidth: 28.4%; Started in v5: 14.6%; Afraid of amount of breaking changes: 12.3%; Already planned but haven't started: 9.5%; Missing incentive: 8.5%; Not a priority: 7.3%; Don't like new styling solution: 4%; Blocked by 3rd party dependencies: 3.3%; Didn't know about it: 3.3%; Migration is in process: 3.3%; Had regressions, therefore came back to v4: 1.3%; Waiting to become stable: 1.3%" />

<p class="blog-description">441 out of 1589 answered</p>

> - **No bandwidth / not priority / blocked by third-party dependencies:** These are all grouped because they are similar in the sense that migration didn’t happen because of a superior force. Not sure what we can do exactly but we put an effort to always have a clear changelog and reasoning for the changes, to equip you with arguments and resources when the time is right to migrate.

Reviewer: ```suggestion
- **No bandwidth / not a priority / blocked by third-party dependencies:** These reasons can all be lumped together as "forces beyond your control." That's to be expected, but we'll continue to do what we can to persuade your team to migrate, and to make the migration as smooth as possible when you're ready.
```

> - **Afraid of the amount of breaking changes:** The fear aspect could probably be attenuated by more documentation, examples, automation, so on and so forth. Things we already covered in previous topics.

Reviewer: I'm sure this isn't what you intended, but the tone here suggests that you're kind of bored with this point. 😅 Let's take it as an opportunity to reiterate our commitment to making improvements instead.
```suggestion
- **Afraid of the amount of breaking changes:** That fear is understandable, and we still have a great deal of work to do with documentation, automation, and other resources to help minimize the friction involved in migrating.
```

> - **Don't like the new styling solution:** We believe that’s totally fair. Not everyone will buy every move. However, as with any open-source project, you always have the opportunity to weigh in and leave your thoughts about any given endeavor we're considering. [The issue discussing the chosen styling solution](https://github.com/mui/material-ui/issues/22342) was fairly big and every contribution added to the safety of deciding on one. So, make sure to participate and contribute if there's any aspect that you don't agree with.

Reviewer: Let's underscore how much we value even critical feedback here.

```suggestion
- **Don't like the new styling solution:** We believe that’s totally fair. Not everyone will buy every move. However, as with any open-source project, you always have the opportunity to weigh in and leave your thoughts about any given endeavor we're considering. [The issue discussing the chosen styling solution](https://github.com/mui/material-ui/issues/22342) was a big one, and every contribution made a difference in the ultimate decision. Please don't hesitate to participate in these open discussions—especially if you see something you disagree with.
```

> - **Didn’t know about it:** There is absolutely an opportunity to explore more channels to engage with the MUI audience, different from those we already use.

Reviewer: Let's be more clear here.

```suggestion
- **Didn’t know about it:** That's something we really need to work on! We'll be exploring new ways to engage with our community in the coming months.
```

> - **Regressions - had to go back to v4:** Feel encouraged to open an issue with whatever difficulty you are having in the migration process, we’ll be happy to guide you.

Reviewer: ```suggestion
- **Regressions - had to go back to v4:** If this happens to you, please open an issue regarding the problems you're facing and we’ll be happy to guide you.
```

> ### Have you, or someone from your team, using any low-code tool for helping with the development of your product?

Reviewer: ```suggestion
### Have you, or anyone from your team, used any low-code tools for helping with the development of your product?
```

<img src="/static/blog/2021-survey/27.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="Yes: 16.1%; No: 83.9%" />

<p class="blog-description">1542 out of 1589 answered</p>

> It’s been a while since we started to play around with ideas on how to broaden the ability to use MUI's component libraries for a wider public, enabling more collaboration and faster development speed. That’s why we had some questions on this year's survey regarding the low-code space. Given that developers are mostly bounded within a product team (considerably tech-literate people), low-code could seem as not necessary, so maybe that’s why over 80% haven’t used it. To expand our knowledge, we had the following questions for the 16% that said yes.

Reviewer: ```suggestion
We see a lot of potential in low-code tools to reach a much wider audience beyond developers, for better collaboration and faster development. Since our existing users are quite tech-literate people, it shouldn't be surprising that over 80% of you are not using low-code tools. But for the 16% who said yes, we had the following questions.
```

### What did it help you build?

<img src="/static/blog/2021-survey/28.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="An internal tool: 22.5%; Landing pages: 22.1%; Analytics dashboard: 17.6%; A design system: 15.6%; E-commerce storefront: 8.6%; A simple customer-facing mobile app: 6.6%; Other: 7%" />

<p class="blog-description">244 out of 1589 answered</p>

> MUI users seem to use low-code tools to build all kinds of applications but mainly for internal tools, landing pages, and analytics dashboards. The findings are similar to the question above where we asked ‘What are you building?’. But it is great to see that there are low-code tools in the market that developers trust. Undoubtedly, it leads to an overall faster feedback and development cycle.

Reviewer: ```suggestion
MUI users are mainly working with low-code tools to build internal tools, landing pages, and analytics dashboards. These findings align with what the rest of our users are building as well. But it is great to see that there are low-code tools in the market that developers trust.
```

> ### If MUI considered building a low-code tool, what primary use case would match your needs?

Reviewer: ```suggestion
### If MUI considered building a low-code tool, what primary use-case would match your needs?
```

<p class="blog-description">1222 out of 1589 answered</p>

### How long have you been developing with JavaScript?

<img src="/static/blog/2021-survey/33.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="Just getting started: 4.8%; 1 year +: 22.2%; 3 years +: 27.9%; 5 years +: 28.2%; 10 years +: 11.1%; 15 years +: 5.9%" />

<p class="blog-description">1520 out of 1589 answered</p>

### How long have you been developing with React?

<img src="/static/blog/2021-survey/34.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="Just getting started: 7%; 6 months +: 11%; 1 year +: 20.6%; 2 years +: 21.2%; 3 years +: 27.1%; 5 years +: 11.5%; Bleeding edge baby! 6 years +: 1.6%" />

<p class="blog-description">1523 out of 1589 answered</p>

### How long have you been developing with MUI?

<img src="/static/blog/2021-survey/35.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="Just getting started: 16.6%; 6 months +: 19.4%; 1 year +: 24.4%; 2 years +: 21.2%; 3 years +: 15.6%; I'm a pioneer! 5 years +: 2.8%" />

<p class="blog-description">1519 out of 1589 answered</p>

## Conclusion

> The annual survey is one of the most important resources we have for deciding on the MUI next steps. We appreciate a lot the time you invested in answering the questions and we can't say enough how invaluable this data is for our team.

Reviewer: Rather than framing it in terms of how great it is for us, we should focus on how much we appreciate our super fans.

```suggestion
The annual survey is one of the most important resources we have for deciding on our next steps. We appreciate all the time you invested in answering our questions, and we can't stress enough how much we value your input.
```

> In 2021, we released the largest MUI major ever. It was an important step towards the future, to continue on the legacy and quality we have always strived for. We also invested resources to consolidate other products we started developing, such as [MUI X](https://mui.com/x/), the [Design kits](https://mui.com/design-kits/), and [premium templates](https://mui.com/templates/). This survey helped to shed light on important topics we need to dig deeper and focus more:

Reviewer: ```suggestion
In 2021, we released the largest update that MUI has ever seen. We also began investing in complementary products, such as [MUI X](https://mui.com/x/), our [design kits](https://mui.com/design-kits/), and [premium templates](https://mui.com/templates/). 

Your feedback from this survey helped us to better understand the areas where we need to improve:
```

> - **Documentation:** You have largely asked for more examples, tutorials, comprehensiveness, and learning material. The docs are probably one of our most important products and making it the best we can is a top priority. With the v5 release, most of 3rd party learning content produced out there about MUI got outdated, so we have a bunch more to expand on official teaching.

Reviewer: ```suggestion
- **Documentation:** You've asked for more examples, more tutorials, more learning materials, and more comprehensive docs. Most third-party content about MUI became outdated with the release of v5, so we have that much more work to do ourselves to pick up the slack.
```

> - **Customization:** A new styling tool was an important step in the right direction to enable more customizability. As with time, fewer and fewer developers rely on Material Design as the default design direction to base their apps on, so the ability and ease of tailoring the components to a certain look become even more important. We’re still continually acknowledging that there are more opportunities to unpack in making customizing MUI components more intuitive.

Reviewer: ```suggestion
- **Customization:** A new styling tool was an important step in the right direction for more customizability. We recognize that there are still many more opportunities to improve the experience of customizing MUI components.
```

> - **Design quality:** Design continues to be the main driving aspect that guides a decision towards a given component library. The number of customization entries reinforces that as well. Additionally, there is a growing number of developers moving away from MUI because of Material Design being the default design direction, and that’s a huge warning sign for us to explore other roads.

Reviewer: ```suggestion
- **Design quality:** Design continues to be the main driving aspect that guides a decision towards a given component library. A growing number of developers are moving away from MUI because of Material Design being the default design direction, and that’s a huge warning sign for us to explore other roads.
```

> - **Breaking changes:** The frequency that developers asked for fewer breaking changes might actually highlight a very positive thing: numerous applications depending on MUI as their component library of choice. To continue supporting your products, we'll definitely look for ways to reduce breaking changes of the following majors. We’ll also continue exploring and deepening our knowledge on automation tools to make it smoother for you when they happen.

Reviewer: ```suggestion
- **Breaking changes:** The need for fewer breaking changes does highlight one very positive point: numerous applications depend on MUI as their component library of choice. To continue supporting your products, we'll definitely look for ways to minimize the number of breaking changes in future updates. We’ll also continue exploring automation tools to ease the pain of any breaking changes that do come up.
```

> - **Commercial vs. MIT balance:** As we have always been an OSS project, offering anything that is paid is bound to cause a sense of friction. However, we saw a lot of requests for more components and features, pointing out that there’s interest in the quality enabled by paid support. We have much more to do to make MUI X components and overall product (pricing model, for example) more mature and compelling. We might as well explore this aspect this year now that we have a full-time team working on it.

Reviewer: Again, I don't think it's really necessary to point out that some people don't like paying for things. 😅 Instead, let's use this as an opportunity to focus on the cool stuff we'll be doing to persuade folks to buy a commercial license.

```suggestion
- **Commercial vs. MIT balance:** MUI will always be an OSS project first and foremost, but we are getting a lot of positive feedback from our paid products and support services, which tells us we're heading in the right direction. We are rapidly expanding our team to better address the needs of all of our users, and we look forward to making both our MIT-licensed and commercial offerings even more robust and compelling.

As we have always been an OSS project, offering anything that is paid is bound to cause a sense of friction. However, we saw a lot of requests for more components and features, pointing out that there’s interest in the quality enabled by paid support. We have much more to do to make MUI X components and overall product (pricing model, for example) more mature and compelling. We might as well explore this aspect this year now that we have a full-time team working on it.
```

> - **Performance:** It is still a bit unclear what ar the low-hanging fruits we could explore regarding performance or what are the biggest hurting problems. But, we're definitely aware of potential places to invest efforts in, like Typescript and MUI System's speed.

Reviewer: ```suggestion
- **Performance:** We're still looking for the most effective ways to optimize performance. TypeScript has potential, and MUI System's speed needs work. We welcome your suggestions!
```

> If you want to continue to influence our roadmap, you can visit [MUI Core](https://github.com/mui/material-ui) and [MUI X](https://github.com/mui/mui-x) repositories, open issues, upvoting the ones you’re most interested in, or leave your comments and impressions about anything else you want to be improved, we'll be happy to hear from you.

Reviewer: ```suggestion
If you want to continue to influence our roadmap, you can visit the [MUI Core](https://github.com/mui/material-ui) and [MUI X](https://github.com/mui/mui-x) repositories, open issues, upvoting the ones you’re most interested in, or leave your comments and impressions about anything else you want to be improved, we'll be happy to hear from you.
```

> Thanks again and until the next survey!

Reviewer: This piece could really use a strong call to action. What is the one thing we want readers to do when they get to the end of this piece?

Reviewer: Yeah, I see where you're coming from. I would suggest that our best option for a CTA here would be to encourage readers to check out our paid products—after all, if you made it all the way to the end, you're probably someone who's open to supporting us financially! 😁

Author: Yeah, we could have something encouraging to either migrate to v5, if they haven't yet or to check MUI X and the Store in general, maybe.

Author: Pushed an iteration now adding a _**Read More**_ section, adding a few relevant to the Survey conversation! Let me know what you think about that.

Reviewer: Excellent idea! Aside from the tiny grammar fixes above, I think this is ready to go. Great job @danilo-leal! It's been a pleasure to collaborate with you on this! 🙌

---
title: 2021 MUI Developer Survey results

> description: Results for our yearly developer survey, 2021 edition.

Reviewer: How can we revise the title and subheader to better describe the value of this post to the reader? I'll do some thinking on this and circle back around with some ideas.

Reviewer: Some ideas for headlines:
- Uncovering new trends in the 2021 MUI developer survey
- What we learned from the 2021 MUI developer survey
- The 2021 MUI developer survey: here's what we discovered

And some subheaders that could go with them:
- Your feedback helps us to build better products. Here's what we learned about your needs in our latest annual survey.
- Find out what we learned from our latest annual user survey, and how we're applying the results to our 2022 roadmap.
- Here's what we learned in our latest survey, and how we're putting your feedback into action.
- You told us what you needed most, and we've heard you loud and clear! Here's how we're putting your feedback into action.

Author: I'm leaning towards: 

> **The 2021 MUI developer survey: here's what we discovered**
> Here's what we learned in our latest survey, and how we're putting your feedback into action.

To share a thought: given that the survey is meant to be a traditional sort of thing, done annually, I believe we should stick to the same title and description across the years. It isn't the usual type of post that we design titles to drive more engagement but rather one that we bet in the consistency aspect (every year we have it) to drive a natural interest in it. There's also the audience aspect to it, as we were discussing in a previous thread. This might be to a very specific, small, audience and not for the general population visiting the docs or even using the library.

Reviewer 4: > we are putting your feedback into action, and in this piece we'll explain how and why.

I agree with the above, however, I would propose that we frame the value to the MUI super-fan at the stage before "putting into action".

The survey is **not** about going into solution mode, I think that this is critically important to keep this marginal. Instead, we try to figure out what are the biggest pain points, what we have improved in the past, what we failed to improve with our actions in the past. It's about 1. synching with reality, 2. bringing transparency to this hence forcing intellectual, 3. and giving direction to the community (and our team) on what matters to be solved next.

So, my proposal here would be to frame it more around you will learn about what the community is experiencing. A s side nice effect is that it show we care about them.

Reviewer: That makes sense @oliviertassinari. In that case, I would lean towards this angle:

title: The 2021 MUI developer survey: here's what we discovered
description: Your feedback helps us to build better products. Here's what we learned about your needs in our annual survey.

Author: Yup, we're all perfectly on the same page. What I was trying to put across is only that across the years, the Survey post should have the same title+description, only with the year changed. I'm good with the above proposal!

Author: Resolving this one.

---
title: 2021 MUI Developer Survey results
description: Results for our yearly developer survey, 2021 edition.
date: 2022-02-22T00:00:00.000Z
authors: ['danilo-leal', 'samuelsycamore', 'oliviertassinari']
tags: ['Developer survey']
---

Keeping up with the tradition, a few months ago we opened the 2021 MUI Developer Survey.
Your feedback helps us to build better products, and we can't thank you enough for being a part of our community.
1,591 of you responded to this year's survey, and we take your input very seriously.
Here's what we've learned about your needs after poring over the results.

Just like the previous iterations, the survey was chopped into three sections: ["Your needs"](#your-needs), ["Your product"](#your-product), and ["About you"](#about-you).

## Your needs

### How would you feel if you could no longer use MUI?

Over 93% of you would be disappointed if you could no longer use MUI (a sum of _"Very disappointed"_ and _“Somewhat disappointed”_).

> That's in keeping with 94% of you who felt the same way when we asked this question in 2019 and 2020.

Author: ```suggestion
That's in keeping with 94% of you who felt the same way when we asked this question in 2019 and 2020. The _"Somewhat disappointed"_ also saw a growth of 9%. These indicators highlight that the evolution of the product that MUI provides might be pointed in the right direction!
```

How does that sound? cc @samuelsycamore

> We'll continue to do our best to keep you satisfied over the long term!

Author: Removed that bit

<img src="/static/blog/2021-survey/1.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="Bar chart: 62.7% Very disappointed, 30.4% Somewhat disappointed, 6.9% Not disappointed." />

<p class="blog-description">1567 out of 1589 answered</p>

Interestingly, the percentage who would be _"Very disappointed"_ to stop using MUI fell by 10%, while those who would be _"Not disappointed"_ grew by 1%.
Those who chose the latter option were asked to explain why.
Here are the biggest reasons we observed:

**The number of available alternatives**

> Lately, with every new sunrise, there's a new UI component library.

Author: Should we remove it or not? cc @samuelsycamore

> Lately, with every new sunrise, there's a new UI component library.

Reviewer 5: Yes

Because they aim to solve the same problems, these libraries often arrive at solutions that are very similar to one another.
The establishment of common industrywide patterns makes it that much more difficult to stand out from the crowd.
We're working hard to expand outward from our core products in order to provide you with tools that our competitors can't match.

**The introduction of a paid product**

> A big challenge with open-source is that the proportion of value captured by maintainers is close to zero.

Author: ```suggestion
A big challenge with open-source is that maintainers capture very little value.
```

Sounds better? cc @samuelsycamore

> A big challenge with open-source is that the proportion of value captured by maintainers is close to zero.

Reviewer: I'm cool with this.

> The MIT licensed model has enabled a lot of growth, but it does have its limitations.

Author: ```suggestion
The MIT licensed model has enabled growing a large community of contributors and the MUI team, but it does have its limitations.
```

Sounds better? cc @samuelsycamore

> The MIT licensed model enabled a lot of growth but it naturally has reached a limit.

Reviewer: In that case, I think I would phrase it something like this:

```suggestion
The MIT licensed model has enabled a lot of growth, but it does have its limitations.
```

> In order to push our mission as much as we believe it's possible, we needed to introduce a way to capture more value.

Author: > To push the products as forward as we believe it's possible, we needed a way to invest more in them.

Sounds better? cc @samuelsycamore

Scale: -100 to 0 needs improvements, 0 to 30 is good, 30 to 70 is great, and 70 to 100 is excellent." />

<p class="blog-description">1563 out of 1589 answered</p>

### What is the main benefit you get from using MUI?

<img src="/static/blog/2021-survey/3.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="This question results in 2019 - Design: 248; Time: 193; DX: 128; Components: 89; Customizability: 53; Documentation: 25; Accessibility: 5; Typescript: 5; Community support: 4; This question results in 2021 - Time: 597; Design: 407; Components: 309; DX: 306; Customizability: 181; Documentation: 57; Community support: 26; Accessibility: 24; Performance: 17" />

<p class="blog-description">1422 out of 1589 answered this question in 2021</p>

Here's what you liked the most about the product in 2021:

- **Community (x2.9):** While appreciation for the community grew by 290%, the community itself doubled in size over that same time frame.
  It looks like we're witnessing a network effect: with twice as many users, we got three times more value from the community.

> - **Accessibility (x2.2):** The community has definitely noticed the work we have put in to make our components more accessible.

Reviewer 4: 2019

  Special thanks to Sebastian Silbermann [(@eps1lon)](https://github.com/eps1lon) for his continued help!

> - **Customizability (x1.6):** Developers seem to be appreciating the new capabilities introduced in v5.

Author: Added in the first sentence:

> Here's what you liked the most about the product in 2021, compared to 2020:

Sounds better?

  That said, we recognize that there is still more work to be done here, given the number of requests we've gotten to better solve this problem.
- **Components (x1.6):** We'll keep adding new components to the library as long as you continue to appreciate them this much!

>   - Component quality (x1.0): You know that we set the bar very high for the quality of our components.

Author: Is there a pattern for this in US English? @samuelsycamore

>   - Component quality (x1.0): You know that we set the bar very high for the quality of our components.

Reviewer: Nothing that I'm aware of. If we want to make it more clear, we could just break the pattern and say "no change" here.

    That can be challenge as we continue to grow, because we expect nothing but the best from our team.
    Thankfully, this result shows that our new members have done an excellent job maintaining our standards.
- **Time (x1.4):** This is an interesting one, because it's not an issue that we specifically addressed.
  Could this just be a result of the growing pressure in the market to build faster? Or because we somehow made this value proposition more noticeable? It could also be related to the next item.
- **Developer experience (x1.1):**

>   - Consistency (x1.75): We didn’t work on this dimension, and yet it grew.

Author: This one:

> What is the main benefit you get from using MUI?

We have categorized consistency as a sub-category of DX.

    It's reasonable to assume that because we have more components than ever before, developers have started to notice how much more consistent their work is when they use MUI.
  - Ease of use (x1.0): This remains unchanged.

>     Could it be because ease of use is mostly defined by the React API (hooks), or is it because we haven't made any major API changes to optimize for ease of use? There could be a lot of work here for our new developer experience team.

Reviewer 4: Developer relations is a subdiscipline of Developer Experience from my perspective.

>     Could it be because ease of use is mostly defined by the React API (hooks), or is it because we haven't made any major API changes to optimize for ease of use? There could be a lot of work here for our new developer experience team.

Reviewer: @mbrookes I like Developer Experience as the umbrella term, drawing from Kurt Kemple's perspective: [Redefining Developer Experience: Why DX is the New DevRel](https://theworst.dev/developer-experience)

>     Could it be because ease of use is mostly defined by the React API (hooks), or is it because we haven't made any major API changes to optimize for ease of use? There could be a lot of work here for our new developer experience team.

Reviewer 4: I love this ressource. I think that it's a matter of how we want to organize the team, knowing that the team organization is a tradeoff that at the end of the day shapes what we build.

IMHO, the DX team should care a lot about shaping the product (experience) for the developers, it would include technical writing, learning, dev rel, marketing. For comparison, I think that the design team should focus on all users, e.g. end-users, with a focus toward using the UI and UX design disiplines to solve problems. The Engineering, DX, and Design departments all gravitates around the product.

<p class="blog-description">1534 out of 1589 answered</p>

It's definitely a good thing to have a general sense of agreement to most of the statements.
However, there's a considerably big gap between _Strongly agree_ and _Agree_ for all but the first statement, which showcases that we need to continue pushing for even more quality in each of these dimensions (customizability, customization, performance, and support).

### What are your most important criteria for choosing a UI library?

<img src="/static/blog/2021-survey/5.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="Ranked list: 1. The design - look and feel; 2. Customizability; 3. Documentation quality; 4. Comprehensiveness; 5. Performance; 6. Popularity; 7. Accessibility; 8. Offered support and help; 9. Bundle size" />

<p class="blog-description">1500 out of 1589 answered - Check <a href="https://www.typeform.com/help/a/ranking-question-360052767651/">Typeform's documentation</a> to understand ranking average calculations</p>

This year we experimented with using Typeform to run the survey, which allows respondents to rank their choices.

> This could've given us a richer perspective than we've been able to capture in the past, but in the end the results weren't too different from last year.

Author: Guess so, but there's also no need to rant on the tool itself.

> This could've given us a richer perspective than we've been able to capture in the past, but in the end the results weren't too different from last year.

Reviewer 4: Yeah right, I didn't see it as a rant. My thought was that for the next person that reads this, he would gain an understanding of the limitations of each tool. Pro & cons.

- **More examples:** Many of the existing learning resources out there are now outdated following the v5 upgrade, so we have a lot of work to do to show you how to make the most of our products.
- **Provide more themes:** Even with the launch of Material v3, Material Design is seen as outdated by many.
  That's why we're working on a second design system that will offer the same high standard of quality as Material, but with an alternative design direction.
- **Fewer breaking changes:** MUI Core v5 definitely introduced some important breaking changes, especially because of the new styling solution.
  Rest assured that we don’t expect to release any major updates this year—in fact, we aim to keep majors at least 12 months apart from each other.
- **Improve customization:** Common requests include making customization easier, providing more examples of common use-cases (like font-family and primary/secondary colors), and improving to theme capabilities.
  Even with the popularity of Emotion and styled-components, there continues to be an enormous need to facilitate customization.

> 

Reviewer 4: I would propose that on top of the recurring topics for improvement (the top existing pains) we also mention the growing pains and the decreasing pain. I did a first attempt at identifying these trends, it could be pushed deeper.

Got ideas for improvements? Please share them with us! Here's how to make sure that your requests get top priority:

- When you create an issue to request features or components in the MUI Core or X repositories, we'll label it as `Waiting for upvotes`.
  The more votes your issue receives, the higher a priority it will become—so make sure to structure, research, and justify your request so it resonates with other community members.
- When requesting a new component, make sure to benchmark existing implementations of it in the wild so we can see what works and what we could better.
- Try to explain the problem you're having as clearly as possible — it's often the case that we already have a component to solve it.
- If you're requesting easier customization, make sure to demonstrate the desired outcome, and explain where you're having trouble in as much detail as you can.
  These requests often highlight opportunities for us to improve our documentation.

> 

Author: I would have the accordion as is, and then leave the Notion page for more in-depth exploration, an internal resource, and not for external reading.

> 

Author: Done.

- **Cheaper Pro plan:** We know it can be out of range for a lot of developers, and that's why the free version is already packed with many features.
  MUI mainly develops open-source software, and we deeply value the OSS ethos of freely sharing what we build so that others can use it and improve upon it.
  - The price of our Pro plan positions it as a product intended for use by teams at professional organizations.
    That said, many who left feedback on the price are individuals, so there might be an opportunity to expand our offerings for those not backed by their company.
- **More features & feature fixes:** Collapsable rows, column resizing, and features for ERP apps were some of the features requested.
  And regarding features to fix, the most common requests were pagination with REST APIs, backend filtering, and cell editor.
- **Improvements to the documentation:** The data grid docs could certainly use a major overhaul—we're working on it.
- **Improvements to the look and feel:** Data grid builders want more default design options beyond Material.
  In terms of UX, many developers mention the filtering experience specifically.

> ### How can we improve the Data Grid Pro for you?

Reviewer 4: It feels weird with the space, I believe we are referring to the JavaScript module here.

```suggestion
### How can we improve the DataGridPro for you?
```

<img src="/static/blog/2021-survey/16.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="More features: 39.8%; Customizability: 20.4%; Fix features: 15.1%; Performance: 5.4%; Cheaper Premium plan: 3.2%; Improve docs: 3.2%; DX - API: 2.2%; LGPL license: 2.2%; More components: 2.2%" />

<p class="blog-description">75 out of 1589 answered</p>

> - **More features:** [Grouping](https://mui.com/components/data-grid/group-pivot/#row-grouping) was the most requested feature for the Data Grid Pro, followed by [master detail](https://mui.com/components/data-grid/group-pivot/#master-detail), aggregation, search, [tree data](https://mui.com/components/data-grid/group-pivot/#tree-data), and [column pinning](https://mui.com/components/data-grid/columns/#column-pinning).

Reviewer 4: ```suggestion
- **More features:** [Grouping](https://mui.com/components/data-grid/group-pivot/#row-grouping) was the most requested feature for the DataGridPro, followed by [master detail](https://mui.com/components/data-grid/group-pivot/#master-detail), aggregation, search, [tree data](https://mui.com/components/data-grid/group-pivot/#tree-data), and [column pinning](https://mui.com/components/data-grid/columns/#column-pinning).
```

  We've already released some of these in the time it took us to analyze the survey results.
  Check the links to see their docs and demos.
  Alternatively, you can follow [our public roadmap](https://github.com/mui/mui-x/projects/1) to see when the others will land.
- **Customizability:** The majority of requests about customizability were asking for possibilities to tweak data grid interactions, which we need to delve into more deeply to better understand.
  Following next were requests for more style customizability, and more documentation on customization.

> - **Fix features:** Filtering got first place as the most requested feature to fix on the Data Grid Pro.

Reviewer 4: The React component

```suggestion
- **Fix features:** Filtering got first place as the most requested feature to fix on the DataGridPro.
```

Reviewer: This goes hand-in-hand with out discussion about when to capitalize components here: https://github.com/mui/material-ui/pull/31101

Author: Yup, let's get to a conclusion there so we don't have the same discussion in different places and then we come back later here with it.

Reviewer 4: > It seems to me that they are one and the same.

@samuelsycamore In this case, I think that we are not referring to the JavaScript module but the feature that comes with the component in this plan. The two are different. The Pro plan of the data grid, as more APIs than `DataGridPro`.

Reviewer 4: ```suggestion
- **Fix features:** Filtering got first place as the most requested feature to fix on the data grid Pro.
```

Reviewer 6: I agree with the conclusion at #31101. 
But as I understand, the "fix features" is directly related to the react component.
- When referring to an MUI component, write it like a React component with backticks - `DataGridPro`

---
title: 'The 2021 MUI developer survey: here's what we discovered'
description: Your feedback helps us to build better products. Here's what we learned about your needs in our annual survey.

> date: 2022-02-22T00:00:00.000Z

Reviewer 4: I guess we can promote it next Monday?

```suggestion
date: 2022-02-28T00:00:00.000Z
```

authors: ['danilo-leal', 'samuelsycamore', 'oliviertassinari']
tags: ['Developer survey']
---

Keeping up with tradition, a few months ago we opened the 2021 MUI Developer Survey.
Your feedback helps us to build better products, and we can't thank you enough for being a part of our community.
1,591 of you responded to this year's survey, and we take your input very seriously.
Here's what we've learned about your needs after poring over the results.

As with the previous iterations, the survey was divided into three sections: ["Your needs"](#your-needs), ["Your product"](#your-product), and ["About you"](#about-you).

## Your needs

### How would you feel if you could no longer use MUI?

Over 93% of you would be disappointed if you could no longer use MUI (a sum of _"Very disappointed"_ and _"Somewhat disappointed"_).

> That's in keeping with 94% of you who felt the same way when we asked this question in 2019 and 2020. The _"Somewhat disappointed"_ also saw a growth of 9%. These indicators highlight that the evolution of the product that MUI provides might be pointed in the right direction!

Reviewer: Let's make the takeaway more direct.

```suggestion
That's in keeping with 94% of you who felt the same way when we asked this question in 2019 and 2020. The _"Somewhat disappointed"_ also saw a growth of 9%. These indicators suggest to us that we're on the right track with our products and their continued evolution.
```

Reviewer 7: I feel like this is a misleading characterization of what happened with "Somewhat disappointed". The "growth" of 9% was entirely a shift from "Very disappointed" to "Somewhat disappointed", so that growth isn't a positive indicator yet the sentence that follows it seems to imply that it **is** a positive change.

Author: Yeah... I think you're right. Maybe a more accurate interpretation would be, starting from the beginning of the answer:

> Over 93% of you would be disappointed if you could no longer use MUI (a sum of "Very disappointed" and "Somewhat disappointed"). That's in keeping with 94% of you who felt the same way when we asked this question in 2019 and 2020. 
>
> However, in comparison with last year, the 9% growth that the "Somewhat disappointed" option saw was most likely from developers who would've had placed their answer in "Very disappointed", since that option had also a 10% drop. Also, those who would be "Not disappointed" grew by 1%. The ones who chose the latter option were asked to explain why, so here are the biggest reasons we observed how that could be:

@samuelsycamore Please advise on the copy here!

Reviewer 4: Agree with @ryancogswell the current wording I see in https://deploy-preview-30999--material-ui.netlify.app/blog/2021-developer-survey-results/#how-would-you-feel-if-you-could-no-longer-use-mui is the opposite of what it should say:

> The "Somewhat disappointed" also saw a growth of 9%. These indicators suggest to us that we're on the right track with our products and their continued evolution.

It's not a growth of 9% of the somewhat disappointed. It's a reduction of 10.6% of the very disappointed (our target). These indicators suggests that we need to course correct, it's getting worse.

We'll continue to do our best to keep you satisfied over the long term!

<img src="/static/blog/2021-survey/1.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="Bar chart: 62.7% Very disappointed, 30.4% Somewhat disappointed, 6.9% Not disappointed." />

<p class="blog-description">1567 out of 1589 answered.</p>

Interestingly, the percentage who would be _"Very disappointed"_ to stop using MUI fell by 10%, while those who would be _"Not disappointed"_ grew by 1%.
Those who chose the latter option were asked to explain why.
Here are the biggest reasons we observed:

**The number of available alternatives**

Lately, with every new sunrise, there's a new UI component library.
Because they aim to solve the same problems, these libraries often arrive at solutions that are very similar to one another.
The establishment of common industrywide patterns makes it that much more difficult to stand out from the crowd.

> We're working hard to expand outward from our core products in order to provide you with tools that our competitors can't match.

Reviewer 4: Same information we fewer words?

```suggestion
We're working hard to expand outward from our core products to provide you with tools that our competitors can't match.
```

**The introduction of a paid product**

A big challenge with open-source is that the proportion of value captured by maintainers is close to zero.
The MIT licensed model has enabled a lot of growth, but it does have its limitations.

> In order to push our mission as much as we believe it's possible, we needed to introduce a way to capture more value.

Reviewer 4: Same information we fewer words?

```suggestion
To push our mission as much as we believe it's possible, we needed to introduce a way to capture more value.
```

> The MIT licensed model has enabled growing a large community of contributors and the MUI team, but it does have its limitations.

Reviewer: ```suggestion
The MIT licensed model has enabled the growth of a large community of contributors as well as the MUI team itself, but it does have its limitations.
```

**The breaking changes made in v5**

Our most recent release, MUI Core v5, introduced some important breaking changes.
The main problem we wanted to tackle was customizability: unlocking more styling options without compromising on performance.

This new styling solution still has a ways to go in terms of maturity, and there's no denying that it requires a considerable amount of work to migrate.
For our part, we still have a lot of work to do to help you make the most of these new features in v5, and we welcome your feedback.

What can we do to help you succeed? If you have any ideas or suggestions, you’re always welcome to share your thoughts by opening an issue on our repository.

### How likely is it that you would recommend MUI to a friend or colleague?

This year we observed a decrease in Net Promoter Score (NPS): from 62 in 2020 to 46 in 2021.
56.4% of you were Promoters this year, down almost 6 percentage points from last year (62.2%).
We're still in a good place — an NPS score between 30 and 70 is considered great.

> But we definitely have much more to do to achieve an excellent score.

Reviewer 4: Same information we fewer words?

```suggestion
But we have much more to do to achieve an excellent score.
```

Scale: -100 to 0 needs improvements, 0 to 30 is good, 30 to 70 is great, and 70 to 100 is excellent." />

<p class="blog-description">1563 out of 1589 answered.</p>

### What is the main benefit you get from using MUI?

<img src="/static/blog/2021-survey/3.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="This question results in 2019 - Design: 248; Time: 193; DX: 128; Components: 89; Customizability: 53; Documentation: 25; Accessibility: 5; Typescript: 5; Community support: 4; This question results in 2021 - Time: 597; Design: 407; Components: 309; DX: 306; Customizability: 181; Documentation: 57; Community support: 26; Accessibility: 24; Performance: 17" />

<p class="blog-description">1422 out of 1589 answered this question in 2021.</p>

Here's what you liked the most about the product in 2021:

- **Community (2.9x):** While appreciation for the community grew by 290%, the community itself doubled in size over that same time frame.
  It looks like we're witnessing a network effect: with twice as many users, we got three times more value from the community.

> - **Accessibility (2.2x):** The community has definitely noticed the work we have put in to make our components more accessible.

Reviewer 4: Same information fewer words?

```suggestion
- **Accessibility (2.2x):** The community has noticed the work we have put in to make our components more accessible.
```

Author: I guess this case benefits from the _"definitely"_ emphasis.

  Special thanks to Sebastian Silbermann [(@eps1lon)](https://github.com/eps1lon) for his continued help!
- **Customizability (1.6x):** Developers seem to be appreciating the new capabilities introduced in v5.
  That said, we recognize that there is still more work to be done here, given the number of requests we've gotten to better solve this problem.
- **Components (1.6x):** We'll keep adding new components to the library as long as you continue to appreciate them this much!
  - Component quality (1.0x): You know that we set the bar very high for the quality of our components.

>     That can be challenge as we continue to grow, because we expect nothing but the best from our team.

Reviewer 4: ```suggestion
    That can be challenging as we continue to grow because we expect nothing but the best from our team.
```

    Thankfully, this result shows that our new members have done an excellent job maintaining our standards.

> - **Time (1.4x):** This is an interesting one, because it's not an issue that we specifically addressed.

Reviewer 4: I guess we don't really need the comma here

```suggestion
- **Time (1.4x):** This is an interesting one because it's not an issue that we specifically addressed.
```

  <tr><th>5</th><th>icons</th><tr>
  <tr><th>3</th><th>typescript</th><tr>
  <tr><th>2</th><th>animations</th><tr>
</table>
</details>

### Please rate how much you agree or disagree with the following statements

<img src="/static/blog/2021-survey/4.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="I can find most of the components I need (Strongly agree: 45.5%, Agree: 45.9%, Neutral: 6.2%, Disagree: 1.5%, Strongly Disagree 0.8%); I can easily customize the components to match the desired design (Strongly agree: 23.4%, Agree: 46.1%, Neutral: 18.6%, Disagree: 9%, Strongly Disagree 2.9%); I can find the answers to most of my questions in the documentation (Strongly agree: 24.1%, Agree: 50.1%, Neutral: 16.3%, Disagree: 7.6%, Strongly Disagree 1.9%); I find the library's performance to be great (Strongly agree: 26.4%, Agree: 44.3%, Neutral: 22.6%, Disagree: 4.7%, Strongly Disagree 2%); Whenever I needed to get help, I received helpful responses, StackOverflow or GitHub (Strongly agree: 20.9%, Agree: 36.7%, Neutral: 36%, Disagree: 4.6%, Strongly Disagree 1.8%)" />

<p class="blog-description">1534 out of 1589 answered.</p>

> It's definitely a good thing to have a general sense of agreement to most of the statements.

Reviewer 4: Fewer words with the same meaning?

```suggestion
It's a good thing to have a general sense of agreement with most of the statements.
```

Author: It seems like _"agreement to"_ is the correct option here ([source](https://www.btb.termiumplus.gc.ca/tpv2guides/guides/wrtps/index-eng.html?lang=eng&lettr=indx_catlog_a&page=9V51F-r5JPyo.html)). But, either way... @samuelsycamore 😅

Reviewer 4: My assumption is "agreement to do something" and "agreement with something"

The notable difference this year seems to be performance climbing its way up, as it has been ranked as the fifth most important aspect.

### What else can we do to improve MUI for you?

<img src="/static/blog/2021-survey/6.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="Word cloud of what we can do to improve MUI" />

<p class="blog-description">1007 out of 1589 answered this question in 2021.</p>

Here are the most recurring topics for improvement:

- **More components:** We continue to see requests for more advanced components such as charts, forms, and calendars, to name a few.
- **More examples:** Many of the existing learning resources out there are now outdated following the v5 upgrade, so we have a lot of work to do to show you how to make the most of our products.
- **Provide more themes:** Even with the launch of Material v3, Material Design is seen as outdated by many.
  That's why we're working on a second design system that will offer the same high standard of quality as Material, but with an alternative design direction.

> - **Fewer breaking changes:** MUI Core v5 definitely introduced some important breaking changes, especially because of the new styling solution.

Reviewer 4: Fewer words with the same meaning?

```suggestion
- **Fewer breaking changes:** MUI Core v5 introduced some important breaking changes, especially because of the new styling solution.
```

  Rest assured that we don’t expect to release any major updates this year—in fact, we aim to keep majors at least 12 months apart from each other.

> - **Improve customization:** Common requests include making customization easier, providing more examples of common use-cases (like font-family and primary/secondary colors), and improving to theme capabilities.

Reviewer 4: A typo?

```suggestion
- **Improve customization:** Common requests include making customization easier, providing more examples of common use-cases (like font-family and primary/secondary colors), and improving the theme capabilities.
```

    <tr><td>1</td><td>fix features - pagination</td><tr>
    <tr><td>1</td><td>fix features - server-side</td><tr>
    <tr><th>5</th><th>docs</th><tr>
    <tr><td>4</td><td>docs - ?</td><tr>
    <tr><td>1</td><td>docs - more examples</td><tr>
    <tr><th>4</th><th>look and feel</th><tr>
    <tr><th>3</th><th>maintain it</th><tr>
    <tr><th>2</th><th>bugs</th><tr>
    <tr><th>2</th><th>DX - MUI Core consistency</th><tr>
    <tr><th>2</th><th>performance</th><tr>
    <tr><th>2</th><th>translation</th><tr>
    <tr><th>1</th><th>a11y</th><tr>
    <tr><th>1</th><th>easier installation</th><tr>
    <tr><th>1</th><th>fewer breaking changes</th><tr>
    <tr><th>1</th><th>hook-only API</th><tr>
    <tr><th>1</th><th>SSR</th><tr>
  </table>
</details>

### How can we improve the Data Grid Pro for you?

<img src="/static/blog/2021-survey/16.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="More features: 39.8%; Customizability: 20.4%; Fix features: 15.1%; Performance: 5.4%; Cheaper Premium plan: 3.2%; Improve docs: 3.2%; DX - API: 2.2%; LGPL license: 2.2%; More components: 2.2%" />

<p class="blog-description">75 out of 1589 answered.</p>

> - **More features:** [Grouping](https://mui.com/components/data-grid/group-pivot/#row-grouping) was the most requested feature for the Data Grid Pro, followed by [master detail](https://mui.com/components/data-grid/group-pivot/#master-detail), aggregation, search, [tree data](https://mui.com/components/data-grid/group-pivot/#tree-data), and [column pinning](https://mui.com/components/data-grid/columns/#column-pinning).

Reviewer 4: Was this the conclusion?

```suggestion
- **More features:** [Grouping](https://mui.com/components/data-grid/group-pivot/#row-grouping) was the most requested feature for the data grid Pro, followed by [master detail](https://mui.com/components/data-grid/group-pivot/#master-detail), aggregation, search, [tree data](https://mui.com/components/data-grid/group-pivot/#tree-data), and [column pinning](https://mui.com/components/data-grid/columns/#column-pinning).
```

Author: @samuelsycamore Can you refresh how are we on this topic?

Reviewer: I think I would leave "the" in here.

It seems reasonable to expect more people to start using type systems.

### Which framework are you using, if any?

<img src="/static/blog/2021-survey/21.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="Create React App: 62.4%; Next.js: 21.9%; Custom webpack: 10.7%; None: 2.6%; Gatsby: 0.9%; Other: 1.5%" />

<p class="blog-description">1497 out of 1589 answered.</p>

This year we saw a considerable bump in Next.js usage, which is not surprising due to its increasing popularity and the advancements it gained in 2021.

### What styling solution are you using?

<img src="/static/blog/2021-survey/22.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="MUI Core v4 (JSS): 45%; Styled components: 37.9%; Emotion: 30.2%; SASS: 20.8%; CSS modules: 18.9%; Vanilla CSS: 17.6%; Tailwind CSS: 9.1%; Stitches: 0.4%; Other: 1.2%" />

<p class="blog-description">1492 out of 1589 answered.</p>

As the MUI Core v5 release is relatively recent, we expect to see many developers still using JSS, and there will probably always be some of you who will prefer it.

> But the growth of Emotion and styled-components here is noteworthy, because they are what our new styling solution is based on.

Reviewer 4: I had asked emotion about the official capitalization of their header, and it seems to be this one https://github.com/emotion-js/emotion

```suggestion
But the growth of emotion and styled-components here is noteworthy because they are what our new styling solution is based on.
```

This year we'll be focusing on expanding and refining [MUI System](https://mui.com/system/basics/) to further improve upon this new styling solution.

### Did you recently migrate to MUI Core v5?

<img src="/static/blog/2021-survey/23.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="Yes: 63.3%; No: 36.7%" />

<p class="blog-description">1546 out of 1589 answered.</p>

> Though many of you who answered the survey have already migrated, we still have a many more to convince to come along with us.

Reviewer 4: a typo?

```suggestion
Though many of you who answered the survey have already migrated, we still have many more to convince to come along with us.
```

That'll be a major focus of ours for the year ahead.

### What's the statement that most defines your migration experience?

<img src="/static/blog/2021-survey/24.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="It was tricky but MUI's documentation and resources helped me to get it right: 43%; It was smooth and it is now working fine: 38.5%; It wasn't good, I had many problems and it took me a long time to finish: 14.3%; It was bad and I ended up regretting it: 2.8%; Other: 1.4%" />

<p class="blog-description">930 out of 1589 answered.</p>

### What could MUI do to improve the migration experience?

> <img src="/static/blog/2021-survey/25.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="General documentation improvements: 43.2%; Automation: 20.7%: Fewer breaking changes: 15.6%; Fewer styling soluiton breaking changes: 8.5%; More tutorials: 5.4%; More styling solution migration tutorial: 4.4%; Migrate Core and X together: 1%" />

Reviewer 4: ```suggestion
<img src="/static/blog/2021-survey/25.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="General documentation improvements: 43.2%; Automation: 20.7%: Fewer breaking changes: 15.6%; Fewer styling solution breaking changes: 8.5%; More tutorials: 5.4%; More styling solution migration tutorial: 4.4%; Migrate Core and X together: 1%" />
```

<p class="blog-description">472 out of 1589 answered.</p>

- **General documentation improvement:** Documentation plays a huge part in defining the developer experience.

>   We definitely need to refine our copywriting, provide more examples, and offer more beginner-friendly resources.

Reviewer 4: Fewer words for the same meaning?

```suggestion
  We need to refine our copywriting, provide more examples, and offer more beginner-friendly resources.
```

Author: I guess this case benefits from the _"definitely"_ emphasis.

Your feedback from this survey helped us to better understand the areas where we need to improve:

- **Documentation:** You've asked for more examples, more tutorials, more learning materials, and more comprehensive docs.
  Most third-party content about MUI became outdated with the release of v5, so we have that much more work to do ourselves to pick up the slack.
- **Customization:** A new styling tool was an important step in the right direction for more customizability.
  We recognize that there are still many more opportunities to improve the experience of customizing MUI components.
- **Design quality:** Design continues to be the main driving aspect that guides a decision towards a given component library.
  A growing number of developers are moving away from MUI because of Material Design being the default design direction, and that’s a huge warning sign for us to explore other roads.
- **Breaking changes:** The need for fewer breaking changes does highlight one very positive point: numerous applications depend on MUI as their component library of choice.

>   To continue supporting your products, we'll definitely look for ways to minimize the number of breaking changes in future updates.

Reviewer 4: Same question as above, do we need this word?

```suggestion
  To continue supporting your products, we'll look for ways to minimize the number of breaking changes in future updates.
```

  We’ll also continue exploring automation tools to ease the pain of any breaking changes that do come up.
- **Commercial vs.
  MIT balance:** MUI will always be an OSS project first and foremost, but we are getting a lot of positive feedback from our paid products and support services, which tells us that we're heading in the right direction.
  We are rapidly expanding our team to better address the needs of all of our users, and we look forward to making both our MIT-licensed and commercial offerings even more robust and compelling.
- **Performance:** We're still looking for the most effective ways to optimize performance.
  TypeScript has potential, and MUI System's speed needs work.
  We welcome your suggestions!

> If you want to continue to influence our roadmap, you can visit the [MUI Core](https://github.com/mui/material-ui) and [MUI X](https://github.com/mui/mui-x) repositories, open issues, upvoting the ones you’re most interested in, or leave your comments and impressions about anything else you want to be improved, we'll be happy to hear from you.

Reviewer: Minor grammar fixes.

```suggestion
If you want to continue to influence our roadmap, you can visit the [MUI Core](https://github.com/mui/material-ui) and [MUI X](https://github.com/mui/mui-x) repositories, open issues, upvote the ones you’re most interested in, or leave your thoughts about anything else you want to be improved. We'd love to hear from you.
```

Thanks again and until the next survey!

### Read more

> Here are related content to some of the topics covered throughout the post:

Reviewer: We can cut this line for the sake of brevity, or else adjust it like so:

```suggestion
Here is some additional content related to the topics covered throughout this post:
```

  We also recognize that we are still lacking in some important documentation around theming and headless APIs.
- **Cheaper Pro plan:** We know it can be out of range for a lot of developers, and that's why the free version is already packed with many features.
  MUI mainly develops open-source software, and we deeply value the OSS ethos of freely sharing what we build so that others can use it and improve upon it.
  - The price of our Pro plan positions it as a product intended for use by teams at professional organizations.
    That said, many who left feedback on the price are individuals, so there might be an opportunity to expand our offerings for those not backed by their company.
- **More features & feature fixes:** Collapsable rows, column resizing, and features for ERP apps were some of the features requested.
  And regarding features to fix, the most common requests were pagination with REST APIs, backend filtering, and cell editor.
- **Improvements to the documentation:** The data grid docs could certainly use a major overhaul—we're working on it.
- **Improvements to the look and feel:** Data grid builders want more default design options beyond Material.

>   In terms of UX, many developers mention the filtering experience specifically.

Reviewer 5: ```suggestion
  In terms of UX improvements, many developers mention the filtering experience specifically.
```

    <tr><td>2</td><td>more features - master detail</td><tr>
    <tr><td>1</td><td>more features - row editing</td><tr>
    <tr><td>1</td><td>more features - column pinning</td><tr>
    <tr><td>1</td><td>more features - column resizing</td><tr>
    <tr><th>6</th><th>fix features</th><tr>
    <tr><td>3</td><td>fix features - filtering</td><tr>
    <tr><td>1</td><td>fix features - editing</td><tr>
    <tr><td>1</td><td>fix features - pagination</td><tr>
    <tr><td>1</td><td>fix features - server-side</td><tr>
    <tr><th>5</th><th>docs</th><tr>
    <tr><td>4</td><td>docs - ?</td><tr>
    <tr><td>1</td><td>docs - more examples</td><tr>
    <tr><th>4</th><th>look and feel</th><tr>
    <tr><th>3</th><th>maintain it</th><tr>
    <tr><th>2</th><th>bugs</th><tr>
    <tr><th>2</th><th>DX - MUI Core consistency</th><tr>
    <tr><th>2</th><th>performance</th><tr>
    <tr><th>2</th><th>translation</th><tr>
    <tr><th>1</th><th>a11y</th><tr>
    <tr><th>1</th><th>easier installation</th><tr>
    <tr><th>1</th><th>fewer breaking changes</th><tr>
    <tr><th>1</th><th>hook-only API</th><tr>
    <tr><th>1</th><th>SSR</th><tr>
  </table>
</details>

> ### How can we improve the Data Grid Pro for you?

Reviewer 5: ```suggestion
### How can we improve Data Grid Pro for you?
```

<img src="/static/blog/2021-survey/16.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="More features: 39.8%; Customizability: 20.4%; Fix features: 15.1%; Performance: 5.4%; Cheaper Premium plan: 3.2%; Improve docs: 3.2%; DX - API: 2.2%; LGPL license: 2.2%; More components: 2.2%" />

<p class="blog-description">75 out of 1589 answered.</p>

- **More features:** [Grouping](https://mui.com/components/data-grid/group-pivot/#row-grouping) was the most requested feature for the Data Grid Pro, followed by [master detail](https://mui.com/components/data-grid/group-pivot/#master-detail), aggregation, search, [tree data](https://mui.com/components/data-grid/group-pivot/#tree-data), and [column pinning](https://mui.com/components/data-grid/columns/#column-pinning).
  We've already released some of these in the time it took us to analyze the survey results.
  Check the links to see their docs and demos.

>   Alternatively, you can follow [our public roadmap](https://github.com/mui/mui-x/projects/1) to see when the others will land.

Reviewer 5: ```suggestion
  Alternatively, you can follow [the public roadmap](https://github.com/mui/mui-x/projects/1) to see when the others will land.
```

- **Customizability:** The majority of requests about customizability were asking for possibilities to tweak data grid interactions, which we need to delve into more deeply to better understand.
  Following next were requests for more style customizability, and more documentation on customization.

> - **Fix features:** Filtering got first place as the most requested feature to fix on the Data Grid Pro.

Reviewer 5: ```suggestion
- **Fix features:** Filtering got first place as the most requested feature to fix on Data Grid Pro.
```

  Following next were lazy loading and server-side rendering support.
- **Performance:** Requests about performance were mostly spread out between runtime and bundle size.
  We must admit that we have put almost no effort into optimizing the bundle size so far.

>   There are likely some low-hanging fruits there.

Reviewer 5: ```suggestion
  There is likely some low-hanging fruit there.
```

Enterprise, dashboard applications, and design systems continue to be in the top 3, similar to last year.
What's changed is that developers are now using MUI more to build e-commerce apps and portfolios, which is pretty cool.
We'd love to see them! Feel free to share in the ["Who's using Material UI?" issue](https://github.com/mui/material-ui/issues/22426).

### What delivery mechanisms are you using?

<img src="/static/blog/2021-survey/19.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="Single-page app (Create React App, etc): 74.7%; Server-side rendered website (Next.js, Gastby, etc): 20.8%; Desktop app (Electron, etc): 3.5%; Native mobile app (Cordova, etc): 0.6%; Other: 0.4%" />

<p class="blog-description">1509 out of 1589 answered.</p>

### What type system are you using?

<img src="/static/blog/2021-survey/20.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="TypeScript: 63.8%; None: 18%; prop-types: 16.6%; Flow: 1.4%; Other: 0.2%" />

<p class="blog-description">1501 out of 1589 answered.</p>

> These results are very different from last year—Typescript just jumped off and it's now the primary type system used out there.

Reviewer 5: ```suggestion
These results are very different from last year—TypeScript took off, and it's now the primary type system used.
```

> It seems reasonable to expect more people to start using type systems.

Reviewer 5: ```suggestion
```

### Which framework are you using, if any?

<img src="/static/blog/2021-survey/21.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="Create React App: 62.4%; Next.js: 21.9%; Custom webpack: 10.7%; None: 2.6%; Gatsby: 0.9%; Other: 1.5%" />

<p class="blog-description">1497 out of 1589 answered.</p>

This year we saw a considerable bump in Next.js usage, which is not surprising due to its increasing popularity and the advancements it gained in 2021.

### What styling solution are you using?

<img src="/static/blog/2021-survey/22.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="MUI Core v4 (JSS): 45%; Styled components: 37.9%; Emotion: 30.2%; SASS: 20.8%; CSS modules: 18.9%; Vanilla CSS: 17.6%; Tailwind CSS: 9.1%; Stitches: 0.4%; Other: 1.2%" />

<p class="blog-description">1492 out of 1589 answered.</p>

> As the MUI Core v5 release is relatively recent, we expect to see many developers still using JSS, and there will probably always be some of you who will prefer it.

Reviewer 5: ```suggestion
As the MUI Core v5 release is relatively recent, we expect to see many developers still using JSS, and there will probably always be some of you who prefer it.
```

But the growth of Emotion and styled-components here is noteworthy, because they are what our new styling solution is based on.
This year we'll be focusing on expanding and refining [MUI System](https://mui.com/system/basics/) to further improve upon this new styling solution.

### Did you recently migrate to MUI Core v5?

<img src="/static/blog/2021-survey/23.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="Yes: 63.3%; No: 36.7%" />

<p class="blog-description">1546 out of 1589 answered.</p>

Though many of you who answered the survey have already migrated, we still have many more to convince to come along with us.

> That'll be a major focus of ours for the year ahead.

Reviewer 5: ```suggestion
That'll be a major focus for the year ahead.
```

In what way?

Author: ```suggestion
We'll strongly focus on documentation refinement and automation tools for the year ahead.
```

Sounds better? cc @samuelsycamore

### What's the statement that most defines your migration experience?

<img src="/static/blog/2021-survey/24.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="It was tricky but MUI's documentation and resources helped me to get it right: 43%; It was smooth and it is now working fine: 38.5%; It wasn't good, I had many problems and it took me a long time to finish: 14.3%; It was bad and I ended up regretting it: 2.8%; Other: 1.4%" />

<p class="blog-description">930 out of 1589 answered.</p>

### What could MUI do to improve the migration experience?

<img src="/static/blog/2021-survey/25.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="General documentation improvements: 43.2%; Automation: 20.7%: Fewer breaking changes: 15.6%; Fewer styling solution breaking changes: 8.5%; More tutorials: 5.4%; More styling solution migration tutorial: 4.4%; Migrate Core and X together: 1%" />

<p class="blog-description">472 out of 1589 answered.</p>

- **General documentation improvement:** Documentation plays a huge part in defining the developer experience.

>   We definitely need to refine our copywriting, provide more examples, and offer more beginner-friendly resources.

Reviewer 5: ```suggestion
  We definitely need to refine our writing, provide more examples, and offer more beginner-friendly resources.
```

- **Automation:** We prepared codemods to help with the v5 migration, given the number of changes.

>   This has been met with positive feedback, which is encouraging since it's our first time providing tools to automate the migration process.

Reviewer 5: ```suggestion
  This has been met with positive feedback, which is encouraging since it's our first time providing comprehensive tools to automate the migration process.
```

Reviewer 5: We provided limited codemods previously.

<p class="blog-description">1523 out of 1589 answered.</p>

### How long have you been developing with MUI?

<img src="/static/blog/2021-survey/35.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="Just getting started: 16.6%; 6 months +: 19.4%; 1 year +: 24.4%; 2 years +: 21.2%; 3 years +: 15.6%; I'm a pioneer! 5 years +: 2.8%" />

<p class="blog-description">1519 out of 1589 answered.</p>

## Conclusion

The annual survey is one of the most important resources we have for deciding on our next steps.
We appreciate all the time you invested in answering our questions, and we can't stress enough how much we value your input.

In 2021, we released the largest update that MUI has ever seen.
We also began investing in complementary products, such as [MUI X](https://mui.com/x/), our [design kits](https://mui.com/design-kits/), and [premium templates](https://mui.com/templates/).

Your feedback from this survey helped us to better understand the areas where we need to improve:

- **Documentation:** You've asked for more examples, more tutorials, more learning materials, and more comprehensive docs.

>   Most third-party content about MUI became outdated with the release of v5, so we have that much more work to do ourselves to pick up the slack.

Reviewer 5: ```suggestion
  Most third-party content about Material UI became outdated with the release of v5, so we have much work to do to pick up the slack.
```

    <tr><td>1</td><td>animations - performance</td><tr>
    <tr><th>7</th><th>fix more bugs</th><tr>
    <tr><th>7</th><th>support other frameworkds (e.g. Vue)</th><tr>
    <tr><th>5</th><th>autocomplete</th><tr>
    <tr><td>3</td><td>autocomplete - ?</td><tr>
    <tr><td>1</td><td>autocomplete - abbreviation</td><tr>
    <tr><td>1</td><td>autocomplete - consistency with text field</td><tr>
    <tr><th>4</th><th>legacy browser support</th><tr>
    <tr><th>3</th><th>components consistency</th><tr>
    <tr><th>2</th><th>less opinionanted</th><tr>
    <tr><th>2</th><th>fix RTL</th><tr>
    <tr><th>2</th><th>class components</th><tr>
  </table>
</details>

## Your product

### What were you primarily using before MUI?

<img src="/static/blog/2021-survey/7.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="Bootstrap: 40.6%; Started with MUI: 37.4%; Tailwind:: 4.8%; Ant Design: 4.8%; Angular Material: 4%; Semantic-UI: 4%; Other: 3.8%; Chakra UI: 0.7%" />

<p class="blog-description">1389 of 1589 answered.</p>

These results are not too different from last year, with the exception that far fewer developers reported using custom systems before MUI.

> It's very encouraging to know that so many of you are building your products with MUI right from the start!

Author: Makes sense — I've removed the excerpt talking about custom systems due to https://github.com/mui/material-ui/pull/30999#discussion_r815461704 so we might be good as far as the post goes. But yeah, it is indeed very much worth keeping this reflection in mind.

When compared with last year's results, client work and personal projects have swapped places this year.
Maybe v5 got developers interested enough to try it on their own sites and apps? It's interesting to see, in any case.

### Who do you collaborate most with?

<img src="/static/blog/2021-survey/10.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="Other developers: 67.2%; Designers: 34.3%; Product managers: 29.2%; No one, it's just me: 28.9%; Other: 0.9%" />

<p class="blog-description">1527 out of 1589 answered.</p>

This is a new question in our annual survey, and the results highlight the fact that our documentation is not just for developers.
It's crucial for our docs—and even the code itself—to be easily understood by designers, product managers, and other less technical team members who may be stakeholders in engineering projects.

### How many web applications did you or your team deliver using MUI this year?

<img src="/static/blog/2021-survey/11.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="0-1: 494; 2-3: 381; 4-5: 122; 6-10: 26; 10+: 28" />

<p class="blog-description">1051 out of 1589 answered.</p>

> Efficacy is one of the toughest metrics to track—how much does MUI actually help developers in shipping their products? The number of apps delivered may vary widely between teams and industries, but it's still encouraging to see that we are enabling developers to build much faster than if they started from scratch.

Author: Right! Changed to an entirely different thing, it wasn't making any sense 😅

<p class="blog-description">1551 out of 1589 answered.</p>

It's not surprising to find out that most of our user base is here for MUI Core.
But it is very encouraging for us to see such a high proportion of MUI X users operating with commercial licenses, and we are excited to continue expanding on our paid products for these users.

There are a lot of opportunities in the MUI X space to unpack.
Let's explore a little bit further below.

### Did you know what MUI X was prior to this survey?

<img src="/static/blog/2021-survey/13.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="Yes: 54.5%; No: 45.5%" />

<p class="blog-description">1312 out of 1589 answered.</p>

A large chunk of you didn’t know what MUI X is prior to the survey, which makes it clear that we still have a lot to do to expand its reach.

### Are you currently using any paid UI component library?

<img src="/static/blog/2021-survey/14.png" style="width: 796px; margin-top: 16px; margin-bottom: 8px;" alt="Yes: 11.8%; No: 88.2%" />

<p class="blog-description">1584 out of 1589 answered.</p>

The results are not very different from last year, aside from a slight increase of respondents saying yes.

> The majority of respondents don’t use any paid libraries, but those who do are mostly using MUI X, which is great news.

Author: Any alternatives to _mission_ or do not use it all?

> The majority of respondents don’t use any paid libraries, but those who do are mostly using MUI X, which is great news.

Reviewer 4: > No business buzzwords please

The higher cause we are pursuing? 😁

> The majority of respondents don’t use any paid libraries, but those who do are mostly using MUI X, which is great news.

Reviewer 4: ```suggestion
The majority of respondents don’t use any paid libraries, but those who do are mostly using MUI X, which validates our assumptions, there is a need to fill that the MIT model is not.
```

We appreciate all the time you invested in answering our questions, and we can't stress enough how much we value your input.

In 2021, we released the largest update that MUI has ever seen.
We also began investing in complementary products, such as [MUI X](https://mui.com/x/), our [design kits](https://mui.com/design-kits/), and [premium templates](https://mui.com/templates/).

Your feedback from this survey helped us to better understand the areas where we need to improve:

- **Documentation:** You've asked for more examples, more tutorials, more learning materials, and more comprehensive docs.
  Most third-party content about MUI became outdated with the release of v5, so we have that much more work to do ourselves to pick up the slack.
- **Customization:** A new styling tool was an important step in the right direction for more customizability.
  We recognize that there are still many more opportunities to improve the experience of customizing MUI components.
- **Design quality:** Design continues to be the main driving aspect that guides a decision towards a given component library.

>   A growing number of developers are moving away from MUI because of Material Design being the default design direction, and that’s a huge warning sign for us to explore other roads.

Author: > not adopting MUI for its Material Design implementation

That implies a bit that there might be a different MD implementation out there better than the one from MUI - and that devs could be going to that one instead of ours. If I were to change the sentence for being even more accurate:

```suggestion
  A growing number of developers are moving away from MUI because of Material Design being the only design direction provided out-of-the-box, and that’s a huge warning sign for us to explore other roads.
```

- **Improve customization:** Common requests include making customization easier, providing more examples of common use-cases (like font-family and primary/secondary colors), and improving the theme capabilities.
  Even with the popularity of emotion and styled-components, there continues to be an enormous need to facilitate customization.

> In comparison with 2020, here are the growing pain points:

Reviewer 4: I think that linking would be enough. It might be better, for curious ones.
